{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- 2 TMT-10plex kits were searched in the same PD study file\n",
    "- Although the data is in one file, they must be treated as separate studies\n",
    "    - P1 = 10plex containing CrAT KO (single knock out) and DKO (double knock out) with floxed controls\n",
    "    - P2 = 10plex containing Sirt3 KO (single knock out) and DKO (double knock out) with floxed controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "peptides_file = 'pd22_exports/pd22_export_Williams_2019_PeptideGroups.txt'\n",
    "proteins_file = 'pd22_exports/pd22_export_Williams_2019_Proteins.txt'\n",
    "pep = pd.read_csv(peptides_file, sep='\\t', low_memory=False)\n",
    "pro = pd.read_csv(proteins_file, sep='\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import data from proteome discoverer"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For the peptides dataframe, use the first listed Master Protein Accession ID as the primary identifier. This identifier will be matched in the Proteins dataframe and used to pull additional information about the peptide. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "pep['Accession'] = pep['Master Protein Accessions'].str.split('; ', expand=True)[0]\n",
    "pro['Entrez'] = pro['Entrez Gene ID'].str.split('; ', expand=True)[0]\n",
    "pro['Ensembl'] = pro['Ensembl Gene ID'].str.split('; ', expand=True)[0]\n",
    "\n",
    "pep = pep.merge(pro[['Accession', 'Description', 'Entrez', 'Gene Symbol']], on='Accession')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, import a pre-generated dataframe (stored as a pickled binary format). The information is based on the MitoCarta2 database, pulled from MitoMiner (http://mitominer.mrc-mbu.cam.ac.uk/release-4.0/begin.do) with additional, manually curated information about the particular location within the mitochondria. This will be used to identify peptides and proteins as mitochondrial.\n",
    "\n",
    "IMS = intermembrane space\n",
    "\n",
    "matrix = mitochondrial matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "mitocarta_df = pd.read_csv('mitocarta.csv')\n",
    "\n",
    "pep['MitoCarta2_List'] = pep.Entrez.isin(mitocarta_df.MouseGeneID.astype(str))\n",
    "pep['Matrix'] = pep.Entrez.isin(mitocarta_df.MouseGeneID[mitocarta_df.Matrix].astype(str))\n",
    "pep['IMS'] = pep.Entrez.isin(mitocarta_df.MouseGeneID[mitocarta_df.IMS].astype(str))\n",
    "\n",
    "pro['MitoCarta2_List'] = pro.Entrez.isin(mitocarta_df.MouseGeneID.astype(str))\n",
    "pro['Matrix'] = pro.Entrez.isin(mitocarta_df.MouseGeneID[mitocarta_df.Matrix].astype(str))\n",
    "pro['IMS'] = pro.Entrez.isin(mitocarta_df.MouseGeneID[mitocarta_df.IMS].astype(str))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, we will identify columns that we want to store in a \"Master Index\". This will contian key information we want to retain from the proteome discoverer 2.2 exported file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "code_folding": [
     1,
     19
    ]
   },
   "outputs": [],
   "source": [
    "# a list to of columns to make a master index\n",
    "protein_mi_list = ['Accession', \n",
    "                   'Description',\n",
    "                   'Entrez',\n",
    "                   'Gene Symbol',\n",
    "                   'Ensembl',\n",
    "                   'MitoCarta2_List',\n",
    "                   'Matrix', \n",
    "                   'IMS',\n",
    "                   'Coverage [%]',\n",
    "                   '# Peptides',\n",
    "                   '# PSMs',\n",
    "                   '# Protein Unique Peptides',\n",
    "                   '# Unique Peptides', \n",
    "                   'Chromosome',\n",
    "                   'Modifications',\n",
    "                   'Acetyl (K) Count',\n",
    "                   'Acetyl (K) Positions']\n",
    "\n",
    "peptide_mi_list = ['Accession', \n",
    "                   'Description',\n",
    "                   'Entrez',\n",
    "                   'Gene Symbol',\n",
    "                   'Modifications', \n",
    "                   'MitoCarta2_List',\n",
    "                   'Matrix', \n",
    "                   'IMS', \n",
    "                   'Modifications in Proteins',\n",
    "                   'Modifications in Proteins (all Sites)',\n",
    "                   'Positions in Proteins',]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, we identify the abundance columns from the experiment. These contain the quantification data for each TMT channel. There are 2 fractions that are mentioned:\n",
    "\n",
    "- F1 = \"Input\", the unenriched, pooled TMT-labeled peptides for all 10 samples. \n",
    "- F2 = \"Acetyl\", the Acetyl-lysine IP-enriched fraction \n",
    "\n",
    "The numbers (126, 130C) indicate the specific TMT tag associated with the column. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "code_folding": [
     0,
     11
    ]
   },
   "outputs": [],
   "source": [
    "p1_acetyl_abundance_list = ['Abundance: F1: 126, Sample, CrATvDKO plex1 (TMT Kit), Acetyl (Fraction), CrATfl_fl (Genotype)',\n",
    "                            'Abundance: F1: 127N, Sample, CrATvDKO plex1 (TMT Kit), Acetyl (Fraction), CrATfl_fl (Genotype)',\n",
    "                            'Abundance: F1: 127C, Sample, CrATvDKO plex1 (TMT Kit), Acetyl (Fraction), CrATfl_fl (Genotype)',\n",
    "                            'Abundance: F1: 128N, Sample, CrATvDKO plex1 (TMT Kit), Acetyl (Fraction), CrATM_min_min (Genotype)',\n",
    "                            'Abundance: F1: 128C, Sample, CrATvDKO plex1 (TMT Kit), Acetyl (Fraction), CrATM_min_min (Genotype)',\n",
    "                            'Abundance: F1: 129N, Sample, CrATvDKO plex1 (TMT Kit), Acetyl (Fraction), CrATM_min_min (Genotype)',\n",
    "                            'Abundance: F1: 129C, Sample, CrATvDKO plex1 (TMT Kit), Acetyl (Fraction), DFC (Genotype)',\n",
    "                            'Abundance: F1: 130N, Sample, CrATvDKO plex1 (TMT Kit), Acetyl (Fraction), DFC (Genotype)',\n",
    "                            'Abundance: F1: 130C, Sample, CrATvDKO plex1 (TMT Kit), Acetyl (Fraction), DKO (Genotype)',\n",
    "                            'Abundance: F1: 131, Sample, CrATvDKO plex1 (TMT Kit), Acetyl (Fraction), DKO (Genotype)',]\n",
    "                            \n",
    "p2_acetyl_abundance_list = ['Abundance: F3: 126, Sample, S3vDKO plex2 (TMT Kit), Acetyl (Fraction), DFC (Genotype)',\n",
    "                            'Abundance: F3: 127N, Sample, S3vDKO plex2 (TMT Kit), Acetyl (Fraction), DFC (Genotype)',\n",
    "                            'Abundance: F3: 127C, Sample, S3vDKO plex2 (TMT Kit), Acetyl (Fraction), DKO (Genotype)',\n",
    "                            'Abundance: F3: 128N, Sample, S3vDKO plex2 (TMT Kit), Acetyl (Fraction), DKO (Genotype)',\n",
    "                            'Abundance: F3: 128C, Sample, S3vDKO plex2 (TMT Kit), Acetyl (Fraction), DKO (Genotype)',\n",
    "                            'Abundance: F3: 129N, Sample, S3vDKO plex2 (TMT Kit), Acetyl (Fraction), Sirt3fl_fl (Genotype)',\n",
    "                            'Abundance: F3: 129C, Sample, S3vDKO plex2 (TMT Kit), Acetyl (Fraction), Sirt3fl_fl (Genotype)',\n",
    "                            'Abundance: F3: 130N, Sample, S3vDKO plex2 (TMT Kit), Acetyl (Fraction), Sirt3M_min_min (Genotype)',\n",
    "                            'Abundance: F3: 130C, Sample, S3vDKO plex2 (TMT Kit), Acetyl (Fraction), Sirt3M_min_min (Genotype)',\n",
    "                            'Abundance: F3: 131, Sample, S3vDKO plex2 (TMT Kit), Acetyl (Fraction), Sirt3M_min_min (Genotype)',]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These are shorter, more friendly versions of those identifier that will be used to rename the columns\n",
    "\n",
    "Genotypes:\n",
    "- S3FL = Sirt3 floxed control\n",
    "- S3KO = Sirt3 knockout (via MCK-Cre)\n",
    "- CrATFL = CrAT floxed control\n",
    "- CrATKO = CrAT knockout (via MCK-Cre)\n",
    "- DFC = Dual flox control (Sirt3 and CrAT floxed)\n",
    "- DKO = Dual Knock-out (Sirt3 and CrAT knockout via MCK-Cre)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "code_folding": [
     0,
     11
    ]
   },
   "outputs": [],
   "source": [
    "p1_acetyl_consolidated_list = ['CrATFL,126,1,Acetyl',\n",
    "                               'CrATFL,127N,1,Acetyl',\n",
    "                               'CrATFL,127C,1,Acetyl',\n",
    "                               'CrATKO,128N,1,Acetyl',\n",
    "                               'CrATKO,128C,1,Acetyl',\n",
    "                               'CrATKO,129N,1,Acetyl',\n",
    "                               'DFC,129C,1,Acetyl',\n",
    "                               'DFC,130N,1,Acetyl',\n",
    "                               'DKO,130C,1,Acetyl',\n",
    "                               'DKO,131,1,Acetyl']\n",
    "                                \n",
    "p2_acetyl_consolidated_list = ['DFC,126,2,Acetyl',\n",
    "                               'DFC,127N,2,Acetyl',\n",
    "                               'DKO,127C,2,Acetyl',\n",
    "                               'DKO,128N,2,Acetyl',\n",
    "                               'DKO,128C,2,Acetyl',\n",
    "                               'S3FL,129N,2,Acetyl',\n",
    "                               'S3FL,129C,2,Acetyl',\n",
    "                               'S3KO,130N,2,Acetyl',\n",
    "                               'S3KO,130C,2,Acetyl',\n",
    "                               'S3KO,131,2,Acetyl']\n",
    "\n",
    "p1_input_consolidated_list = [x.replace('Acetyl', 'Input') for x in p1_acetyl_consolidated_list]\n",
    "p2_input_consolidated_list = [x.replace('Acetyl', 'Input') for x in p2_acetyl_consolidated_list]\n",
    "\n",
    "p1_input_abundance_list = [x.replace('Acetyl', 'Input').replace('F1', 'F2') for x in p1_acetyl_abundance_list]\n",
    "p2_input_abundance_list = [x.replace('Acetyl', 'Input').replace('F3', 'F4') for x in p2_acetyl_abundance_list]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1_acetyl_columns_df = pd.DataFrame.from_records([col.split(',') for col in p1_acetyl_consolidated_list])\n",
    "p2_acetyl_columns_df = pd.DataFrame.from_records([col.split(',') for col in p2_acetyl_consolidated_list])\n",
    "\n",
    "p1_input_columns_df = pd.DataFrame.from_records([col.split(',') for col in p1_input_consolidated_list])\n",
    "p2_input_columns_df = pd.DataFrame.from_records([col.split(',') for col in p2_input_consolidated_list])\n",
    "\n",
    "p1_acetyl_columns_df.set_index(p1_acetyl_columns_df.columns.tolist(), inplace=True)\n",
    "p2_acetyl_columns_df.set_index(p2_acetyl_columns_df.columns.tolist(), inplace=True)\n",
    "\n",
    "p1_input_columns_df.set_index(p1_input_columns_df.columns.tolist(), inplace=True)\n",
    "p2_input_columns_df.set_index(p2_input_columns_df.columns.tolist(), inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Locate the Acetyl Peptides (Kac)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We look for peptides that were quantified from the Acetyl fraction AND contain an identified actyl-lysine residue present. While the acetyl fraction is enriched for acetylated lysine residues, not all identified peptides within this fraction contain this post-translational modification. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# acetyl peptides\n",
    "p1_acetyl_df = pep[pep['Modifications'].str.contains('Acetyl')][p1_acetyl_abundance_list].dropna()\n",
    "p2_acetyl_df = pep[pep['Modifications'].str.contains('Acetyl')][p2_acetyl_abundance_list].dropna()\n",
    "\n",
    "p1_acetyl_df.columns = p1_acetyl_columns_df.index\n",
    "p2_acetyl_df.columns = p2_acetyl_columns_df.index\n",
    "\n",
    "# Flatten the column headers for working in R\n",
    "p1_acetyl_df.columns = ['_'.join(col).strip() for col in p1_acetyl_df.columns.values]\n",
    "p2_acetyl_df.columns = ['_'.join(col).strip() for col in p2_acetyl_df.columns.values]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the input - i.e., everything identified in the non-enriched fraction\n",
    "p1_input_df = pep[p1_input_abundance_list].dropna()\n",
    "p2_input_df = pep[p2_input_abundance_list].dropna()\n",
    "\n",
    "p1_input_df.columns = p1_input_columns_df.index\n",
    "p2_input_df.columns = p2_input_columns_df.index\n",
    "\n",
    "# Flatten the column headers for working in R\n",
    "p1_input_df.columns = ['_'.join(col).strip() for col in p1_input_df.columns.values]\n",
    "p2_input_df.columns = ['_'.join(col).strip() for col in p2_input_df.columns.values]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Locate the protein abundances\n",
    "- Use values obtained from the input fraction - these represent the total proteome\n",
    "- Only select proteins that are called \"Master Proteins\" by proteome discoverer 2.2 wiht a FDR < 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1_protein_df = (pro[(pro.Master=='IsMasterProtein')\n",
    "                  &(pro['Exp. q-value: Combined']<=0.01)]\n",
    "              [p1_input_abundance_list]\n",
    "              .dropna(how='all')\n",
    "              .copy())\n",
    "\n",
    "p1_protein_df.columns = p1_input_df.columns\n",
    "p1_protein_df.rename(columns=lambda x: x.replace('Input', 'Proteins'), inplace=True)\n",
    "\n",
    "p2_protein_df = (pro[(pro.Master=='IsMasterProtein')\n",
    "                  &(pro['Exp. q-value: Combined']<=0.01)]\n",
    "              [p2_input_abundance_list]\n",
    "              .dropna(how='all')\n",
    "              .copy())\n",
    "\n",
    "p2_protein_df.columns = p2_input_df.columns\n",
    "p2_protein_df.rename(columns=lambda x: x.replace('Input', 'Proteins'), inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Normalize Data\n",
    "- Correct for variations in total protein loading per kit\n",
    "- use the mean of the sums of total peptide loading (input fraction) to generate a correction factor\n",
    "- apply this correction factor to the acetyl peptides, 'input fraction', and proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculate the scaling factor for each plex "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1_channel_sums = p1_input_df.sum(axis=0)\n",
    "p2_channel_sums = p2_input_df.sum(axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plex 1: Individual Channel, Loading relative to the mean\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CrATFL_126_1_Input     0.796103\n",
       "CrATFL_127N_1_Input    1.062434\n",
       "CrATFL_127C_1_Input    1.102380\n",
       "CrATKO_128N_1_Input    1.024336\n",
       "CrATKO_128C_1_Input    0.995921\n",
       "CrATKO_129N_1_Input    0.988315\n",
       "DFC_129C_1_Input       1.038378\n",
       "DFC_130N_1_Input       0.985814\n",
       "DKO_130C_1_Input       0.956391\n",
       "DKO_131_1_Input        1.049928\n",
       "dtype: float64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print('Plex 1: Individual Channel, Loading relative to the mean')\n",
    "\n",
    "p1_channel_sums.divide(p1_channel_sums.mean())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plex 1: Scaling factor to adjust relative loading to 1\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CrATFL_126_1_Input     1.256118\n",
       "CrATFL_127N_1_Input    0.941235\n",
       "CrATFL_127C_1_Input    0.907128\n",
       "CrATKO_128N_1_Input    0.976243\n",
       "CrATKO_128C_1_Input    1.004096\n",
       "CrATKO_129N_1_Input    1.011824\n",
       "DFC_129C_1_Input       0.963041\n",
       "DFC_130N_1_Input       1.014390\n",
       "DKO_130C_1_Input       1.045597\n",
       "DKO_131_1_Input        0.952446\n",
       "dtype: float64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print('Plex 1: Scaling factor to adjust relative loading to 1')\n",
    "\n",
    "p1_scaling_factor = 1/p1_channel_sums.divide(p1_channel_sums.mean())\n",
    "p1_scaling_factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plex 2: Individual Channel, Loading relative to the mean\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "DFC_126_2_Input      0.928235\n",
       "DFC_127N_2_Input     1.083377\n",
       "DKO_127C_2_Input     1.100762\n",
       "DKO_128N_2_Input     1.069166\n",
       "DKO_128C_2_Input     0.973270\n",
       "S3FL_129N_2_Input    0.950931\n",
       "S3FL_129C_2_Input    1.034328\n",
       "S3KO_130N_2_Input    0.958474\n",
       "S3KO_130C_2_Input    0.981884\n",
       "S3KO_131_2_Input     0.919571\n",
       "dtype: float64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print('Plex 2: Individual Channel, Loading relative to the mean')\n",
    "\n",
    "p2_channel_sums.divide(p2_channel_sums.mean())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plex 2: Scaling factor to adjust relative loading to 1\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "DFC_126_2_Input      1.077313\n",
       "DFC_127N_2_Input     0.923039\n",
       "DKO_127C_2_Input     0.908461\n",
       "DKO_128N_2_Input     0.935309\n",
       "DKO_128C_2_Input     1.027465\n",
       "S3FL_129N_2_Input    1.051601\n",
       "S3FL_129C_2_Input    0.966811\n",
       "S3KO_130N_2_Input    1.043325\n",
       "S3KO_130C_2_Input    1.018450\n",
       "S3KO_131_2_Input     1.087463\n",
       "dtype: float64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print('Plex 2: Scaling factor to adjust relative loading to 1')\n",
    "\n",
    "p2_scaling_factor = 1/p2_channel_sums.divide(p2_channel_sums.mean())\n",
    "p2_scaling_factor"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Apply the Scaling Factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CrATFL_126_1_Input_norm     3029072.32\n",
       "CrATFL_127N_1_Input_norm    3029072.32\n",
       "CrATFL_127C_1_Input_norm    3029072.32\n",
       "CrATKO_128N_1_Input_norm    3029072.32\n",
       "CrATKO_128C_1_Input_norm    3029072.32\n",
       "CrATKO_129N_1_Input_norm    3029072.32\n",
       "DFC_129C_1_Input_norm       3029072.32\n",
       "DFC_130N_1_Input_norm       3029072.32\n",
       "DKO_130C_1_Input_norm       3029072.32\n",
       "DKO_131_1_Input_norm        3029072.32\n",
       "dtype: float64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p1_input_norm = (p1_input_df * p1_scaling_factor)\n",
    "p1_input_norm.columns = [str(col) + '_norm' for col in p1_input_norm.columns]\n",
    "p1_input_norm.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DFC_126_2_Input_norm      2835985.02\n",
       "DFC_127N_2_Input_norm     2835985.02\n",
       "DKO_127C_2_Input_norm     2835985.02\n",
       "DKO_128N_2_Input_norm     2835985.02\n",
       "DKO_128C_2_Input_norm     2835985.02\n",
       "S3FL_129N_2_Input_norm    2835985.02\n",
       "S3FL_129C_2_Input_norm    2835985.02\n",
       "S3KO_130N_2_Input_norm    2835985.02\n",
       "S3KO_130C_2_Input_norm    2835985.02\n",
       "S3KO_131_2_Input_norm     2835985.02\n",
       "dtype: float64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p2_input_norm = (p2_input_df * p2_scaling_factor)\n",
    "p2_input_norm.columns = [str(col) + '_norm' for col in p2_input_norm.columns]\n",
    "p2_input_norm.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CrATFL_126_1_Proteins_norm     1.801427e+06\n",
       "CrATFL_127N_1_Proteins_norm    1.812383e+06\n",
       "CrATFL_127C_1_Proteins_norm    1.805524e+06\n",
       "CrATKO_128N_1_Proteins_norm    1.777052e+06\n",
       "CrATKO_128C_1_Proteins_norm    1.813146e+06\n",
       "CrATKO_129N_1_Proteins_norm    1.783128e+06\n",
       "DFC_129C_1_Proteins_norm       1.812656e+06\n",
       "DFC_130N_1_Proteins_norm       1.785270e+06\n",
       "DKO_130C_1_Proteins_norm       1.806918e+06\n",
       "DKO_131_1_Proteins_norm        1.797056e+06\n",
       "dtype: float64"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p1_protein_norm = (p1_protein_df * p1_scaling_factor.rename(lambda x: x.replace('Input', 'Proteins')))\n",
    "p1_protein_norm.columns = [str(col) + '_norm' for col in p1_protein_norm.columns]\n",
    "p1_protein_norm.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DFC_126_2_Proteins_norm      1.654504e+06\n",
       "DFC_127N_2_Proteins_norm     1.663691e+06\n",
       "DKO_127C_2_Proteins_norm     1.692797e+06\n",
       "DKO_128N_2_Proteins_norm     1.658630e+06\n",
       "DKO_128C_2_Proteins_norm     1.697719e+06\n",
       "S3FL_129N_2_Proteins_norm    1.656868e+06\n",
       "S3FL_129C_2_Proteins_norm    1.711811e+06\n",
       "S3KO_130N_2_Proteins_norm    1.661136e+06\n",
       "S3KO_130C_2_Proteins_norm    1.649256e+06\n",
       "S3KO_131_2_Proteins_norm     1.657424e+06\n",
       "dtype: float64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p2_protein_norm = (p2_protein_df * p2_scaling_factor.rename(lambda x: x.replace('Input', 'Proteins')))\n",
    "p2_protein_norm.columns = [str(col) + '_norm' for col in p2_protein_norm.columns]\n",
    "p2_protein_norm.sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The intensities for the acetyl peptides will not sum to the same number after this correction factor - this is because there are different levels of acetylation in the genotypes. You can see that the Genetic knock-outs have ↑ signal relative their flox control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CrATFL_126_1_Acetyl_norm      82407.140456\n",
       "CrATFL_127N_1_Acetyl_norm     76833.319507\n",
       "CrATFL_127C_1_Acetyl_norm     80878.155561\n",
       "CrATKO_128N_1_Acetyl_norm     97116.707060\n",
       "CrATKO_128C_1_Acetyl_norm    101603.344537\n",
       "CrATKO_129N_1_Acetyl_norm     93492.503420\n",
       "DFC_129C_1_Acetyl_norm        83985.925779\n",
       "DFC_130N_1_Acetyl_norm        77553.166664\n",
       "DKO_130C_1_Acetyl_norm       153635.212301\n",
       "DKO_131_1_Acetyl_norm        136605.685611\n",
       "dtype: float64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p1_acetyl_norm = p1_acetyl_df * p1_scaling_factor.rename(lambda x: x.replace('Input', 'Acetyl'))\n",
    "p1_acetyl_norm.columns = [str(col) + '_norm' for col in p1_acetyl_norm.columns]\n",
    "p1_acetyl_norm.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DFC_126_2_Acetyl_norm       50134.603739\n",
       "DFC_127N_2_Acetyl_norm      51292.740622\n",
       "DKO_127C_2_Acetyl_norm     127262.067606\n",
       "DKO_128N_2_Acetyl_norm     105069.564430\n",
       "DKO_128C_2_Acetyl_norm     115084.761966\n",
       "S3FL_129N_2_Acetyl_norm     42498.438985\n",
       "S3FL_129C_2_Acetyl_norm     51771.657244\n",
       "S3KO_130N_2_Acetyl_norm     55096.749688\n",
       "S3KO_130C_2_Acetyl_norm     57722.696542\n",
       "S3KO_131_2_Acetyl_norm      52647.349156\n",
       "dtype: float64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p2_acetyl_norm = p2_acetyl_df * p2_scaling_factor.rename(lambda x: x.replace('Input', 'Acetyl'))\n",
    "p2_acetyl_norm.columns = [str(col) + '_norm' for col in p2_acetyl_norm.columns]\n",
    "p2_acetyl_norm.sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Calculate the \"relative occupancy\" for each plex\n",
    "- This adjusts the acetyl peptide quantification based on changes in protein quantification. This corrects not for experimental loading, but for genetic changes that may occur between the genetic knock-out and the flox control.\n",
    "- The relative occupancy corrections will be almost un-noticed in these animals, however. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:7: RuntimeWarning: divide by zero encountered in log2\n",
      "  import sys\n"
     ]
    }
   ],
   "source": [
    "# convert the input-normalized data into the log2 space\n",
    "# subtract the row mean from each column to center each row around 0 (in log 2)\n",
    "p1_acetyl_norm_log2   = np.log2(p1_acetyl_norm).sub(np.log2(p1_acetyl_norm).mean(axis=1), axis='index')\n",
    "p1_input_norm_log2    = np.log2(p1_input_norm).sub(np.log2(p1_input_norm).mean(axis=1), axis='index')\n",
    "p1_protein_norm_log2  = np.log2(p1_protein_norm).sub(np.log2(p1_protein_norm).mean(axis=1), axis='index')\n",
    "\n",
    "p2_acetyl_norm_log2   = np.log2(p2_acetyl_norm).sub(np.log2(p2_acetyl_norm).mean(axis=1), axis='index')\n",
    "p2_input_norm_log2    = np.log2(p2_input_norm).sub(np.log2(p2_input_norm).mean(axis=1), axis='index')\n",
    "p2_protein_norm_log2  = np.log2(p2_protein_norm).sub(np.log2(p2_protein_norm).mean(axis=1), axis='index')\n",
    "\n",
    "\n",
    "# rename columns to save the information\n",
    "p1_acetyl_norm_log2.columns  = [str(col) + '_log2' for col in p1_acetyl_norm_log2.columns]\n",
    "p1_input_norm_log2.columns   = [str(col) + '_log2' for col in p1_input_norm_log2.columns]\n",
    "p1_protein_norm_log2.columns = [str(col) + '_log2' for col in p1_protein_norm_log2.columns]\n",
    "\n",
    "p2_acetyl_norm_log2.columns  = [str(col) + '_log2' for col in p2_acetyl_norm_log2.columns]\n",
    "p2_input_norm_log2.columns   = [str(col) + '_log2' for col in p2_input_norm_log2.columns]\n",
    "p2_protein_norm_log2.columns = [str(col) + '_log2' for col in p2_protein_norm_log2.columns]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Combine the the \"raw\", load-normalized, and log2-transformed data into a single dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# plex 1\n",
    "p1_acetyl = pd.concat([pep.loc[p1_acetyl_norm.index][peptide_mi_list], \n",
    "                       p1_acetyl_df, \n",
    "                       p1_acetyl_norm, \n",
    "                       p1_acetyl_norm_log2,], \n",
    "                      axis=1)\n",
    "\n",
    "p1_input = pd.concat([pep.loc[p1_input_norm.index][peptide_mi_list], \n",
    "                      p1_input_df, \n",
    "                      p1_input_norm, \n",
    "                      p1_input_norm_log2], \n",
    "                     axis=1)\n",
    "\n",
    "p1_protein = pd.concat([pro.loc[p1_protein_df.index][protein_mi_list], \n",
    "                        p1_protein_df, \n",
    "                        p1_protein_norm, \n",
    "                        p1_protein_norm_log2], \n",
    "                       axis=1)\n",
    "\n",
    "# plex2\n",
    "p2_acetyl = pd.concat([pep.loc[p2_acetyl_norm.index][peptide_mi_list], \n",
    "                       p2_acetyl_df, \n",
    "                       p2_acetyl_norm, \n",
    "                       p2_acetyl_norm_log2,], \n",
    "                      axis=1)\n",
    "\n",
    "p2_input = pd.concat([pep.loc[p2_input_norm.index][peptide_mi_list], \n",
    "                      p2_input_df, \n",
    "                      p2_input_norm, \n",
    "                      p2_input_norm_log2], \n",
    "                     axis=1)\n",
    "\n",
    "p2_protein = pd.concat([pro.loc[p2_protein_df.index][protein_mi_list], \n",
    "                        p2_protein_df, \n",
    "                        p2_protein_norm, \n",
    "                        p2_protein_norm_log2], \n",
    "                       axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculate the RO\n",
    "- Merge protein information along with acetyl information into a single dataframe\n",
    "- The protein abundance information will be used to correc the acetyl data (the relative occupancy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1_protein_merge_cols = p1_protein.filter(regex='_log2').columns.tolist() + ['Accession']\n",
    "p2_protein_merge_cols = p2_protein.filter(regex='_log2').columns.tolist() + ['Accession']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1_merged_acetyl = p1_acetyl.merge(p1_protein[p1_protein_merge_cols], \n",
    "                                   on='Accession', \n",
    "                                   how='left', \n",
    "                                   validate='m:1')\n",
    "\n",
    "p2_merged_acetyl = p2_acetyl.merge(p2_protein[p2_protein_merge_cols], \n",
    "                                   on='Accession', \n",
    "                                   how='left', \n",
    "                                   validate='m:1')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The relative occupancy of the post-translational modification is calculated by subtracting changes in the protien abundance from the acetyl-peptide abundance. For example, if a PTM site is found to be increasing by 2 fold, but the protein abundnace is also increasing by 2 fold, then the relative occupancy would be 0. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CrATFL_126_1_Acetyl_norm_log2_ro</th>\n",
       "      <th>CrATFL_127N_1_Acetyl_norm_log2_ro</th>\n",
       "      <th>CrATFL_127C_1_Acetyl_norm_log2_ro</th>\n",
       "      <th>CrATKO_128N_1_Acetyl_norm_log2_ro</th>\n",
       "      <th>CrATKO_128C_1_Acetyl_norm_log2_ro</th>\n",
       "      <th>CrATKO_129N_1_Acetyl_norm_log2_ro</th>\n",
       "      <th>DFC_129C_1_Acetyl_norm_log2_ro</th>\n",
       "      <th>DFC_130N_1_Acetyl_norm_log2_ro</th>\n",
       "      <th>DKO_130C_1_Acetyl_norm_log2_ro</th>\n",
       "      <th>DKO_131_1_Acetyl_norm_log2_ro</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>-0.228927</td>\n",
       "      <td>-0.397147</td>\n",
       "      <td>-0.430871</td>\n",
       "      <td>-0.171143</td>\n",
       "      <td>-0.068982</td>\n",
       "      <td>0.235369</td>\n",
       "      <td>-0.013883</td>\n",
       "      <td>0.582442</td>\n",
       "      <td>0.045253</td>\n",
       "      <td>0.447889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-0.064612</td>\n",
       "      <td>-0.413235</td>\n",
       "      <td>-0.386901</td>\n",
       "      <td>-0.077529</td>\n",
       "      <td>-0.448411</td>\n",
       "      <td>0.235798</td>\n",
       "      <td>-0.194681</td>\n",
       "      <td>0.441240</td>\n",
       "      <td>0.466737</td>\n",
       "      <td>0.441594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>-0.500649</td>\n",
       "      <td>-0.527844</td>\n",
       "      <td>-0.044728</td>\n",
       "      <td>0.013550</td>\n",
       "      <td>-0.424221</td>\n",
       "      <td>0.130864</td>\n",
       "      <td>-0.369122</td>\n",
       "      <td>0.680328</td>\n",
       "      <td>0.784873</td>\n",
       "      <td>0.256949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>-0.142318</td>\n",
       "      <td>-0.359298</td>\n",
       "      <td>-0.391600</td>\n",
       "      <td>0.082026</td>\n",
       "      <td>-0.261988</td>\n",
       "      <td>0.064649</td>\n",
       "      <td>-0.338449</td>\n",
       "      <td>0.675071</td>\n",
       "      <td>0.197805</td>\n",
       "      <td>0.474103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>-0.239486</td>\n",
       "      <td>-0.607886</td>\n",
       "      <td>0.017914</td>\n",
       "      <td>0.547402</td>\n",
       "      <td>-0.433459</td>\n",
       "      <td>0.139333</td>\n",
       "      <td>0.155283</td>\n",
       "      <td>0.713918</td>\n",
       "      <td>-0.149299</td>\n",
       "      <td>-0.143720</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CrATFL_126_1_Acetyl_norm_log2_ro  CrATFL_127N_1_Acetyl_norm_log2_ro  \\\n",
       "0                         -0.228927                          -0.397147   \n",
       "1                         -0.064612                          -0.413235   \n",
       "2                         -0.500649                          -0.527844   \n",
       "3                         -0.142318                          -0.359298   \n",
       "4                         -0.239486                          -0.607886   \n",
       "\n",
       "   CrATFL_127C_1_Acetyl_norm_log2_ro  CrATKO_128N_1_Acetyl_norm_log2_ro  \\\n",
       "0                          -0.430871                          -0.171143   \n",
       "1                          -0.386901                          -0.077529   \n",
       "2                          -0.044728                           0.013550   \n",
       "3                          -0.391600                           0.082026   \n",
       "4                           0.017914                           0.547402   \n",
       "\n",
       "   CrATKO_128C_1_Acetyl_norm_log2_ro  CrATKO_129N_1_Acetyl_norm_log2_ro  \\\n",
       "0                          -0.068982                           0.235369   \n",
       "1                          -0.448411                           0.235798   \n",
       "2                          -0.424221                           0.130864   \n",
       "3                          -0.261988                           0.064649   \n",
       "4                          -0.433459                           0.139333   \n",
       "\n",
       "   DFC_129C_1_Acetyl_norm_log2_ro  DFC_130N_1_Acetyl_norm_log2_ro  \\\n",
       "0                       -0.013883                        0.582442   \n",
       "1                       -0.194681                        0.441240   \n",
       "2                       -0.369122                        0.680328   \n",
       "3                       -0.338449                        0.675071   \n",
       "4                        0.155283                        0.713918   \n",
       "\n",
       "   DKO_130C_1_Acetyl_norm_log2_ro  DKO_131_1_Acetyl_norm_log2_ro  \n",
       "0                        0.045253                       0.447889  \n",
       "1                        0.466737                       0.441594  \n",
       "2                        0.784873                       0.256949  \n",
       "3                        0.197805                       0.474103  \n",
       "4                       -0.149299                      -0.143720  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate relative occupancy in acetyl peptides\n",
    "# p1\n",
    "p1_relative_occupancy = (p1_merged_acetyl.filter(regex='Acetyl_norm_log2')\n",
    "                         .sub(p1_merged_acetyl.filter(regex='Proteins_norm_log2')\n",
    "                              # rename the protein column headers to enable easy dataframe subtraction\n",
    "                              .rename(columns=lambda x: x.replace('Proteins', 'Acetyl'))))\n",
    "\n",
    "p1_relative_occupancy.rename(columns=lambda x: x+'_ro', inplace=True)\n",
    "\n",
    "p1_relative_occupancy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DFC_126_2_Acetyl_norm_log2_ro</th>\n",
       "      <th>DFC_127N_2_Acetyl_norm_log2_ro</th>\n",
       "      <th>DKO_127C_2_Acetyl_norm_log2_ro</th>\n",
       "      <th>DKO_128N_2_Acetyl_norm_log2_ro</th>\n",
       "      <th>DKO_128C_2_Acetyl_norm_log2_ro</th>\n",
       "      <th>S3FL_129N_2_Acetyl_norm_log2_ro</th>\n",
       "      <th>S3FL_129C_2_Acetyl_norm_log2_ro</th>\n",
       "      <th>S3KO_130N_2_Acetyl_norm_log2_ro</th>\n",
       "      <th>S3KO_130C_2_Acetyl_norm_log2_ro</th>\n",
       "      <th>S3KO_131_2_Acetyl_norm_log2_ro</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>-1.020246</td>\n",
       "      <td>-0.372482</td>\n",
       "      <td>1.015874</td>\n",
       "      <td>1.300822</td>\n",
       "      <td>1.238225</td>\n",
       "      <td>-1.448226</td>\n",
       "      <td>-0.824477</td>\n",
       "      <td>0.024452</td>\n",
       "      <td>0.108021</td>\n",
       "      <td>-0.021963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.197794</td>\n",
       "      <td>0.466177</td>\n",
       "      <td>-0.275599</td>\n",
       "      <td>0.123803</td>\n",
       "      <td>0.143688</td>\n",
       "      <td>0.413960</td>\n",
       "      <td>-1.323232</td>\n",
       "      <td>-0.263887</td>\n",
       "      <td>-0.061468</td>\n",
       "      <td>-0.421236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>-0.073142</td>\n",
       "      <td>-0.065819</td>\n",
       "      <td>-0.016426</td>\n",
       "      <td>0.267592</td>\n",
       "      <td>0.051071</td>\n",
       "      <td>0.005400</td>\n",
       "      <td>-1.184937</td>\n",
       "      <td>0.363596</td>\n",
       "      <td>0.602540</td>\n",
       "      <td>0.050127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.661939</td>\n",
       "      <td>0.421147</td>\n",
       "      <td>-0.106358</td>\n",
       "      <td>0.367535</td>\n",
       "      <td>0.294433</td>\n",
       "      <td>0.110343</td>\n",
       "      <td>-1.364902</td>\n",
       "      <td>-0.480644</td>\n",
       "      <td>0.669652</td>\n",
       "      <td>-0.573143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.753240</td>\n",
       "      <td>0.693138</td>\n",
       "      <td>-0.001591</td>\n",
       "      <td>1.022336</td>\n",
       "      <td>-0.454821</td>\n",
       "      <td>-0.389432</td>\n",
       "      <td>-0.410803</td>\n",
       "      <td>-0.208252</td>\n",
       "      <td>-0.410613</td>\n",
       "      <td>-0.593201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   DFC_126_2_Acetyl_norm_log2_ro  DFC_127N_2_Acetyl_norm_log2_ro  \\\n",
       "0                      -1.020246                       -0.372482   \n",
       "1                       1.197794                        0.466177   \n",
       "2                      -0.073142                       -0.065819   \n",
       "3                       0.661939                        0.421147   \n",
       "4                       0.753240                        0.693138   \n",
       "\n",
       "   DKO_127C_2_Acetyl_norm_log2_ro  DKO_128N_2_Acetyl_norm_log2_ro  \\\n",
       "0                        1.015874                        1.300822   \n",
       "1                       -0.275599                        0.123803   \n",
       "2                       -0.016426                        0.267592   \n",
       "3                       -0.106358                        0.367535   \n",
       "4                       -0.001591                        1.022336   \n",
       "\n",
       "   DKO_128C_2_Acetyl_norm_log2_ro  S3FL_129N_2_Acetyl_norm_log2_ro  \\\n",
       "0                        1.238225                        -1.448226   \n",
       "1                        0.143688                         0.413960   \n",
       "2                        0.051071                         0.005400   \n",
       "3                        0.294433                         0.110343   \n",
       "4                       -0.454821                        -0.389432   \n",
       "\n",
       "   S3FL_129C_2_Acetyl_norm_log2_ro  S3KO_130N_2_Acetyl_norm_log2_ro  \\\n",
       "0                        -0.824477                         0.024452   \n",
       "1                        -1.323232                        -0.263887   \n",
       "2                        -1.184937                         0.363596   \n",
       "3                        -1.364902                        -0.480644   \n",
       "4                        -0.410803                        -0.208252   \n",
       "\n",
       "   S3KO_130C_2_Acetyl_norm_log2_ro  S3KO_131_2_Acetyl_norm_log2_ro  \n",
       "0                         0.108021                       -0.021963  \n",
       "1                        -0.061468                       -0.421236  \n",
       "2                         0.602540                        0.050127  \n",
       "3                         0.669652                       -0.573143  \n",
       "4                        -0.410613                       -0.593201  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate relative occupancy in acetyl peptides\n",
    "\n",
    "p2_relative_occupancy = (p2_merged_acetyl.filter(regex='Acetyl_norm_log2')\n",
    "                         .sub(p2_merged_acetyl.filter(regex='Proteins_norm_log2')\n",
    "                              # rename the protein column headers to enable easy dataframe subtraction\n",
    "                              .rename(columns=lambda x: x.replace('Proteins', 'Acetyl'))))\n",
    "\n",
    "p2_relative_occupancy.rename(columns=lambda x: x+'_ro', inplace=True)\n",
    "\n",
    "p2_relative_occupancy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# stick the relative occupancy data onto the merged data\n",
    "p1_merged_acetyl = pd.concat([p1_merged_acetyl, p1_relative_occupancy], axis=1)\n",
    "p2_merged_acetyl = pd.concat([p2_merged_acetyl, p2_relative_occupancy], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop the protein columns out of the data\n",
    "p1_merged_acetyl.drop(columns=p1_merged_acetyl.filter(regex='_Protein').columns.tolist(), \n",
    "                      inplace=True)\n",
    "\n",
    "p2_merged_acetyl.drop(columns=p2_merged_acetyl.filter(regex='_Protein').columns.tolist(), \n",
    "                      inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# *limma* analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Export Files for *limma* analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It is probably just as easy to do this entire analysis in R, with the limma component incorporated into the data processing. I'm not very experienced in R, so the disjointed analysis with R and python was faster for me. Exporting the data from Python into a CSV to import into R was the quickest method for my purposes. One could also try the rpy2 library (I tinkered with it, but found it to be too much effort for the payoff) or exporting the dataframes from pandas into a feather/arrow format."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1_acetyl_norm.to_csv('limma/p1_limma_export_kac.csv')\n",
    "p2_acetyl_norm.to_csv('limma/p2_limma_export_kac.csv')\n",
    "\n",
    "p1_merged_acetyl.filter(regex='_ro$').to_csv('limma/p1_limma_export_RO_kac.csv')\n",
    "p2_merged_acetyl.filter(regex='_ro$').to_csv('limma/p2_limma_export_RO_kac.csv')\n",
    "\n",
    "p1_protein.filter(regex='.+_norm$').to_csv('limma/p1_limma_export_pro.csv')\n",
    "p2_protein.filter(regex='.+_norm$').to_csv('limma/p2_limma_export_pro.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read Files from *limma* analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1_kac_limma = pd.read_csv('limma/p1_eb_fit_kac.tsv', sep='\\t')\n",
    "p2_kac_limma = pd.read_csv('limma/p2_eb_fit_kac.tsv', sep='\\t')\n",
    "\n",
    "p1_kac_ro_limma = pd.read_csv('limma/p1_eb_fit_RO_kac.tsv', sep='\\t')\n",
    "p2_kac_ro_limma = pd.read_csv('limma/p2_eb_fit_RO_kac.tsv', sep='\\t')\n",
    "\n",
    "p1_prot_limma = pd.read_csv('limma/p1_eb_fit_pro.tsv', sep='\\t')\n",
    "p2_prot_limma = pd.read_csv('limma/p2_eb_fit_pro.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge back onto data\n",
    "p1_acetyl = pd.concat([p1_acetyl, p1_kac_limma], axis=1)\n",
    "p2_acetyl = pd.concat([p2_acetyl, p2_kac_limma], axis=1)\n",
    "\n",
    "p1_protein = pd.concat([p1_protein, p1_prot_limma], axis=1)\n",
    "p2_protein = pd.concat([p2_protein, p2_prot_limma], axis=1)\n",
    "\n",
    "p1_ro_acetyl = pd.concat([p1_merged_acetyl, p1_kac_ro_limma], axis=1)\n",
    "p2_ro_acetyl = pd.concat([p2_merged_acetyl, p2_kac_ro_limma], axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Rename the limma results to have more intuitive names\n",
    "\n",
    "- `Coef` = Log2 Fold Change comparison between two groups\n",
    "- `Res` = Significance. \n",
    "    - 0 = not significant\n",
    "    - 1 = significant difference increasing\n",
    "    - -1 = significant difference decreasing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "limma_match = ['Coef\\.', 'p\\.value\\.', 't\\.', 'Res\\.']\n",
    "limma_replace = ['_Log2FC', '_p_value', '_t', '_significant']\n",
    "\n",
    "for match, replace in zip(limma_match, limma_replace):\n",
    "    p1_acetyl.rename(columns=lambda x: re.sub(match + '(.+)', '\\g<1>' + replace, x), inplace=True)\n",
    "    p1_protein.rename(columns=lambda x: re.sub(match + '(.+)', '\\g<1>' + replace, x), inplace=True)\n",
    "    p1_ro_acetyl.rename(columns=lambda x: re.sub(match + '(.+)', '\\g<1>' + replace, x), inplace=True)\n",
    "    \n",
    "    p2_acetyl.rename(columns=lambda x: re.sub(match + '(.+)', '\\g<1>' + replace, x), inplace=True)\n",
    "    p2_protein.rename(columns=lambda x: re.sub(match + '(.+)', '\\g<1>' + replace, x), inplace=True)\n",
    "    p2_ro_acetyl.rename(columns=lambda x: re.sub(match + '(.+)', '\\g<1>' + replace, x), inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pull out useful information from the Description field into separate columns. Although there is a Gene Symbol field present, it isn't always populated and isn't always the same"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1_acetyl['Name'] = p1_acetyl.Description.str.split(' [OSGNPESV]{2}=', expand=True)[0]\n",
    "p1_acetyl['Symbol'] = p1_acetyl.Description.str.split(' [OSGNPESV]{2}=', expand=True)[2]\n",
    "\n",
    "p1_protein['Name'] = p1_protein.Description.str.split(' [OSGNPESV]{2}=', expand=True)[0]\n",
    "p1_protein['Symbol'] = p1_protein.Description.str.split(' [OSGNPESV]{2}=', expand=True)[2]\n",
    "\n",
    "p1_ro_acetyl['Name'] = p1_ro_acetyl.Description.str.split(' [OSGNPESV]{2}=', expand=True)[0]\n",
    "p1_ro_acetyl['Symbol'] = p1_ro_acetyl.Description.str.split(' [OSGNPESV]{2}=', expand=True)[2]\n",
    "\n",
    "# reorder the columns so the information\n",
    "p1_acetyl = p1_acetyl[p1_acetyl.columns.tolist()[-2:] + p1_acetyl.columns.tolist()[:-2]]\n",
    "p1_protein = p1_protein[p1_protein.columns.tolist()[-2:] + p1_protein.columns.tolist()[:-2]]\n",
    "p1_ro_acetyl = p1_ro_acetyl[p1_ro_acetyl.columns.tolist()[-2:] + p1_ro_acetyl.columns.tolist()[:-2]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "p2_acetyl['Name'] = p2_acetyl.Description.str.split(' [OSGNPESV]{2}=', expand=True)[0]\n",
    "p2_acetyl['Symbol'] = p2_acetyl.Description.str.split(' [OSGNPESV]{2}=', expand=True)[2]\n",
    "\n",
    "p2_protein['Name'] = p2_protein.Description.str.split(' [OSGNPESV]{2}=', expand=True)[0]\n",
    "p2_protein['Symbol'] = p2_protein.Description.str.split(' [OSGNPESV]{2}=', expand=True)[2]\n",
    "\n",
    "p2_ro_acetyl['Name'] = p2_ro_acetyl.Description.str.split(' [OSGNPESV]{2}=', expand=True)[0]\n",
    "p2_ro_acetyl['Symbol'] = p2_ro_acetyl.Description.str.split(' [OSGNPESV]{2}=', expand=True)[2]\n",
    "\n",
    "# reorder the columns so the information\n",
    "p2_acetyl = p2_acetyl[p2_acetyl.columns.tolist()[-2:] + p2_acetyl.columns.tolist()[:-2]]\n",
    "p2_protein = p2_protein[p2_protein.columns.tolist()[-2:] + p2_protein.columns.tolist()[:-2]]\n",
    "p2_ro_acetyl = p2_ro_acetyl[p2_ro_acetyl.columns.tolist()[-2:] + p2_ro_acetyl.columns.tolist()[:-2]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sort the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1_protein.sort_values('Acetyl (K) Count', ascending=False, inplace=True)\n",
    "p2_protein.sort_values('Acetyl (K) Count', ascending=False, inplace=True)\n",
    "\n",
    "p1_acetyl.sort_values('DKOvsDFC_Log2FC', ascending=False, inplace=True)\n",
    "p2_acetyl.sort_values('DKOvsDFC_Log2FC', ascending=False, inplace=True)\n",
    "\n",
    "p1_ro_acetyl.sort_values('DKOvsDFC_Log2FC', ascending=False, inplace=True)\n",
    "p2_ro_acetyl.sort_values('DKOvsDFC_Log2FC', ascending=False, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Top 25 Fold Changes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is one (simple) method to assess the severity of hyperacetylation. We'll only look at the relative occupancy corrected acetylpeptide abundances for this analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First, we will limit our analysis to the mitochondria as defined by the mitocarta2 dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1 = p1_ro_acetyl[p1_ro_acetyl.MitoCarta2_List==True].copy()\n",
    "p2 = p2_ro_acetyl[p2_ro_acetyl.MitoCarta2_List==True].copy()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, we will move out of the log2 space for more intuitive comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "p1['DKOvsDFC_FC'] = np.exp2(p1.DKOvsDFC_Log2FC)\n",
    "p1['CrATKOvsCrATFL_FC'] = np.exp2(p1.CrATKOvsCrATFL_Log2FC)\n",
    "\n",
    "p2['DKOvsDFC_FC'] = np.exp2(p2.DKOvsDFC_Log2FC)\n",
    "p2['S3KOvsS3FL_FC'] = np.exp2(p2.S3KOvsS3FL_Log2FC)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, we want to pull out the specific lysine residues acetylated on each individual peptide. We want these residue positions related back to the master protein (i.e., not the position within the peptide). This isn't easily achieved in how the PD data is currently exported. Some peptides could correspond to multiple protein accessions, each with different lysine positions. All possible accessions and positions are listed in the peptide output - however, the order is arbitrary. Therefor, we need to iterate though the accessions listed and match them back to the master protein accession. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>919</th>\n",
       "      <td>Q61425 1xAcetyl [K312]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1143</th>\n",
       "      <td>Q9CZU6 1xAcetyl [K393]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>764</th>\n",
       "      <td>Q8K2B3 1xAcetyl [K179]</td>\n",
       "      <td>A0A1Y7VJ55 1xAcetyl [K177]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>550</th>\n",
       "      <td>Q8BMS1 1xAcetyl [K334]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1145</th>\n",
       "      <td>Q9CZU6 1xAcetyl [K52]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           0                           1     2     3     4  \\\n",
       "919   Q61425 1xAcetyl [K312]                        None  None  None  None   \n",
       "1143  Q9CZU6 1xAcetyl [K393]                        None  None  None  None   \n",
       "764   Q8K2B3 1xAcetyl [K179]  A0A1Y7VJ55 1xAcetyl [K177]  None  None  None   \n",
       "550   Q8BMS1 1xAcetyl [K334]                        None  None  None  None   \n",
       "1145   Q9CZU6 1xAcetyl [K52]                        None  None  None  None   \n",
       "\n",
       "         5     6     7     8     9  \n",
       "919   None  None  None  None  None  \n",
       "1143  None  None  None  None  None  \n",
       "764   None  None  None  None  None  \n",
       "550   None  None  None  None  None  \n",
       "1145  None  None  None  None  None  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p1_kac_sites = p1['Modifications in Proteins'].str.split('; (?!K)', expand=True, )\n",
    "p1_kac_sites.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>752</th>\n",
       "      <td>Q9CZU6 1xAcetyl [K52]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>661</th>\n",
       "      <td>Q8BWT1 1xAcetyl [K209]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>750</th>\n",
       "      <td>Q9CZU6 1xAcetyl [K393]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>609</th>\n",
       "      <td>Q9WUR2 1xAcetyl [K60]</td>\n",
       "      <td>E9PUY9 1xAcetyl [K27]</td>\n",
       "      <td>E9PVM6 1xAcetyl [K27]</td>\n",
       "      <td>Q3TCD4 1xAcetyl [K47]</td>\n",
       "      <td>E9PYA9 1xAcetyl [K27]</td>\n",
       "      <td>E9Q858 1xAcetyl [K27]</td>\n",
       "      <td>Q9WUR2-2 1xAcetyl [K27]</td>\n",
       "      <td>E9PYC6 1xAcetyl [K27]</td>\n",
       "      <td>E9Q7A8 1xAcetyl [K27]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>P08249 1xAcetyl [K239]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>Q8BMS1 1xAcetyl [K259]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>812</th>\n",
       "      <td>P03930 1xAcetyl [K46]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>Q8BMS1 2xAcetyl [K411; K413]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>Q8BMS1 1xAcetyl [K]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>768</th>\n",
       "      <td>F7D3P8 1xAcetyl [K31]</td>\n",
       "      <td>Q9DB20 1xAcetyl [K97]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               0                      1   \\\n",
       "752         Q9CZU6 1xAcetyl [K52]                   None   \n",
       "661        Q8BWT1 1xAcetyl [K209]                   None   \n",
       "750        Q9CZU6 1xAcetyl [K393]                   None   \n",
       "609         Q9WUR2 1xAcetyl [K60]  E9PUY9 1xAcetyl [K27]   \n",
       "341        P08249 1xAcetyl [K239]                   None   \n",
       "280        Q8BMS1 1xAcetyl [K259]                   None   \n",
       "812         P03930 1xAcetyl [K46]                   None   \n",
       "301  Q8BMS1 2xAcetyl [K411; K413]                   None   \n",
       "286           Q8BMS1 1xAcetyl [K]                   None   \n",
       "768         F7D3P8 1xAcetyl [K31]  Q9DB20 1xAcetyl [K97]   \n",
       "\n",
       "                        2                      3                      4   \\\n",
       "752                   None                   None                   None   \n",
       "661                   None                   None                   None   \n",
       "750                   None                   None                   None   \n",
       "609  E9PVM6 1xAcetyl [K27]  Q3TCD4 1xAcetyl [K47]  E9PYA9 1xAcetyl [K27]   \n",
       "341                   None                   None                   None   \n",
       "280                   None                   None                   None   \n",
       "812                   None                   None                   None   \n",
       "301                   None                   None                   None   \n",
       "286                   None                   None                   None   \n",
       "768                   None                   None                   None   \n",
       "\n",
       "                        5                        6                      7   \\\n",
       "752                   None                     None                   None   \n",
       "661                   None                     None                   None   \n",
       "750                   None                     None                   None   \n",
       "609  E9Q858 1xAcetyl [K27]  Q9WUR2-2 1xAcetyl [K27]  E9PYC6 1xAcetyl [K27]   \n",
       "341                   None                     None                   None   \n",
       "280                   None                     None                   None   \n",
       "812                   None                     None                   None   \n",
       "301                   None                     None                   None   \n",
       "286                   None                     None                   None   \n",
       "768                   None                     None                   None   \n",
       "\n",
       "                        8     9     10  \n",
       "752                   None  None  None  \n",
       "661                   None  None  None  \n",
       "750                   None  None  None  \n",
       "609  E9Q7A8 1xAcetyl [K27]  None  None  \n",
       "341                   None  None  None  \n",
       "280                   None  None  None  \n",
       "812                   None  None  None  \n",
       "301                   None  None  None  \n",
       "286                   None  None  None  \n",
       "768                   None  None  None  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p2_kac_sites = p2['Modifications in Proteins'].str.split('; (?!K)', expand=True, )\n",
    "p2_kac_sites.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In the above data, peptides with multiple accessions can have the master protein accession listed in any of the 10 columns above - it isn't always in the first column. Therefore, we will iterate through the columns of these sites to match them back to the original accession. The matches will be combined into a list, and the list will be transformed into a Series"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# list to hold the sites associated with the same master protein accession\n",
    "p1_sites = []\n",
    "for i in np.arange(0, p1_kac_sites.columns[-1]+1):\n",
    "    bool_mask = p1.Accession == p1_kac_sites[i].str.split(' ', expand=True)[0]\n",
    "    try:\n",
    "        temp_sites = p1_kac_sites[i].loc[bool_mask].str.split('(?<!;)\\s', expand=True)[2]\n",
    "        p1_sites.append(temp_sites)\n",
    "    except:\n",
    "        continue\n",
    "        \n",
    "p2_sites = []\n",
    "for i in np.arange(0, p2_kac_sites.columns[-1]+1):\n",
    "    bool_mask = p2.Accession == p2_kac_sites[i].str.split(' ', expand=True)[0]\n",
    "    try:\n",
    "        temp_sites = p2_kac_sites[i].loc[bool_mask].str.split('(?<!;)\\s', expand=True)[2]\n",
    "        p2_sites.append(temp_sites)\n",
    "    except:\n",
    "        continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "919           K312\n",
       "1143          K393\n",
       "764           K179\n",
       "550           K334\n",
       "1145           K52\n",
       "1142      K49; K52\n",
       "616     K531; K539\n",
       "836           K260\n",
       "946            K60\n",
       "1093          K226\n",
       "Name: Acetylated Residues, dtype: object"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# convert the list into a pandas series     \n",
    "p1_sites = pd.concat(p1_sites)\n",
    "\n",
    "# remove the percent confidence and brackets \n",
    "p1_sites = p1_sites.str.replace('\\(\\d{1,6}\\)', '').str.replace('\\[(.+)\\]', '\\g<1>')\n",
    "\n",
    "p1_sites.rename('Acetylated Residues', inplace=True)\n",
    "\n",
    "p1_sites.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "752           K52\n",
       "661          K209\n",
       "750          K393\n",
       "609           K60\n",
       "341          K239\n",
       "280          K259\n",
       "812           K46\n",
       "301    K411; K413\n",
       "286             K\n",
       "196          K283\n",
       "Name: Acetylated Residues, dtype: object"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# convert the list into a pandas series     \n",
    "p2_sites = pd.concat(p2_sites)\n",
    "\n",
    "# remove the percent confidence and brackets \n",
    "p2_sites = p2_sites.str.replace('\\(\\d{1,6}\\)', '').str.replace('\\[(.+)\\]', '\\g<1>')\n",
    "\n",
    "p2_sites.rename('Acetylated Residues', inplace=True)\n",
    "\n",
    "p2_sites.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, we add the specific sites back to the dataframe and prepare to export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge the residues back onto the dataframe\n",
    "p1 = p1.merge(p1_sites.to_frame(), left_index=True, right_index=True)\n",
    "p2 = p2.merge(p2_sites.to_frame(), left_index=True, right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>Name</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Entrez</th>\n",
       "      <th>Matrix</th>\n",
       "      <th>IMS</th>\n",
       "      <th>Acetylated Residues</th>\n",
       "      <th>DKOvsDFC_FC</th>\n",
       "      <th>CrATKOvsCrATFL_FC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>919</th>\n",
       "      <td>Q61425</td>\n",
       "      <td>Hydroxyacyl-coenzyme A dehydrogenase, mitochon...</td>\n",
       "      <td>Hadh</td>\n",
       "      <td>Hadh</td>\n",
       "      <td>15107</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K312</td>\n",
       "      <td>16.546091</td>\n",
       "      <td>1.302894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1143</th>\n",
       "      <td>Q9CZU6</td>\n",
       "      <td>Citrate synthase, mitochondrial</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Cs</td>\n",
       "      <td>12974</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K393</td>\n",
       "      <td>16.359824</td>\n",
       "      <td>1.418364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>764</th>\n",
       "      <td>Q8K2B3</td>\n",
       "      <td>Succinate dehydrogenase [ubiquinone] flavoprot...</td>\n",
       "      <td>Sdha</td>\n",
       "      <td>Sdha</td>\n",
       "      <td>66945</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K179</td>\n",
       "      <td>16.164221</td>\n",
       "      <td>1.199551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>550</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K334</td>\n",
       "      <td>14.830350</td>\n",
       "      <td>1.352968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1145</th>\n",
       "      <td>Q9CZU6</td>\n",
       "      <td>Citrate synthase, mitochondrial</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Cs</td>\n",
       "      <td>12974</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K52</td>\n",
       "      <td>14.824442</td>\n",
       "      <td>1.795159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1142</th>\n",
       "      <td>Q9CZU6</td>\n",
       "      <td>Citrate synthase, mitochondrial</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Cs</td>\n",
       "      <td>12974</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K49; K52</td>\n",
       "      <td>14.108287</td>\n",
       "      <td>1.378787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616</th>\n",
       "      <td>Q03265</td>\n",
       "      <td>ATP synthase subunit alpha, mitochondrial</td>\n",
       "      <td>Atp5a1</td>\n",
       "      <td>Atp5a1</td>\n",
       "      <td>11946</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K531; K539</td>\n",
       "      <td>13.955865</td>\n",
       "      <td>2.165616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>Q8QZT1</td>\n",
       "      <td>Acetyl-CoA acetyltransferase, mitochondrial</td>\n",
       "      <td>Acat1</td>\n",
       "      <td>Acat1</td>\n",
       "      <td>110446</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K260</td>\n",
       "      <td>13.620694</td>\n",
       "      <td>1.284992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>946</th>\n",
       "      <td>Q9WUR2</td>\n",
       "      <td>Enoyl-CoA delta isomerase 2, mitochondrial</td>\n",
       "      <td>Eci2</td>\n",
       "      <td>Eci2</td>\n",
       "      <td>23986</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K60</td>\n",
       "      <td>11.757587</td>\n",
       "      <td>1.287511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>Q99LC5</td>\n",
       "      <td>Electron transfer flavoprotein subunit alpha, ...</td>\n",
       "      <td>Etfa</td>\n",
       "      <td>Etfa</td>\n",
       "      <td>110842</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K226</td>\n",
       "      <td>11.376252</td>\n",
       "      <td>1.483240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1027</th>\n",
       "      <td>Q8BWT1</td>\n",
       "      <td>3-ketoacyl-CoA thiolase, mitochondrial</td>\n",
       "      <td>Acaa2</td>\n",
       "      <td>Acaa2</td>\n",
       "      <td>52538</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K209</td>\n",
       "      <td>11.061172</td>\n",
       "      <td>1.586879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>Q8BFR5</td>\n",
       "      <td>Elongation factor Tu, mitochondrial</td>\n",
       "      <td>Tufm</td>\n",
       "      <td>Tufm</td>\n",
       "      <td>233870</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K88</td>\n",
       "      <td>10.965843</td>\n",
       "      <td>1.561059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>Q9DCM2</td>\n",
       "      <td>Glutathione S-transferase kappa 1</td>\n",
       "      <td>Gstk1</td>\n",
       "      <td>Gstk1</td>\n",
       "      <td>76263</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>K93</td>\n",
       "      <td>10.913403</td>\n",
       "      <td>1.171825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>555</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K406</td>\n",
       "      <td>10.145445</td>\n",
       "      <td>1.121315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>P42125</td>\n",
       "      <td>Enoyl-CoA delta isomerase 1, mitochondrial</td>\n",
       "      <td>Eci1</td>\n",
       "      <td>Eci1</td>\n",
       "      <td>13177</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K255</td>\n",
       "      <td>9.797140</td>\n",
       "      <td>2.060486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>P08249</td>\n",
       "      <td>Malate dehydrogenase, mitochondrial</td>\n",
       "      <td>Mdh2</td>\n",
       "      <td>Mdh2</td>\n",
       "      <td>17448</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K239</td>\n",
       "      <td>9.716530</td>\n",
       "      <td>1.409407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>Q8K4F5</td>\n",
       "      <td>Protein ABHD11</td>\n",
       "      <td>Abhd11</td>\n",
       "      <td>Abhd11</td>\n",
       "      <td>68758</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K196</td>\n",
       "      <td>9.649334</td>\n",
       "      <td>1.307799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>626</th>\n",
       "      <td>Q03265</td>\n",
       "      <td>ATP synthase subunit alpha, mitochondrial</td>\n",
       "      <td>Atp5a1</td>\n",
       "      <td>Atp5a1</td>\n",
       "      <td>11946</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K498</td>\n",
       "      <td>8.830434</td>\n",
       "      <td>1.179997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1144</th>\n",
       "      <td>Q9CZU6</td>\n",
       "      <td>Citrate synthase, mitochondrial</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Cs</td>\n",
       "      <td>12974</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K459</td>\n",
       "      <td>8.363489</td>\n",
       "      <td>1.307401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1222</th>\n",
       "      <td>P03930</td>\n",
       "      <td>ATP synthase protein 8</td>\n",
       "      <td>Mtatp8</td>\n",
       "      <td>ATP8</td>\n",
       "      <td>17706</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>K46</td>\n",
       "      <td>8.320988</td>\n",
       "      <td>1.230055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>P42125</td>\n",
       "      <td>Enoyl-CoA delta isomerase 1, mitochondrial</td>\n",
       "      <td>Eci1</td>\n",
       "      <td>Eci1</td>\n",
       "      <td>13177</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K283</td>\n",
       "      <td>7.993041</td>\n",
       "      <td>1.210816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>969</th>\n",
       "      <td>Q9Z2I9</td>\n",
       "      <td>Succinate--CoA ligase [ADP-forming] subunit be...</td>\n",
       "      <td>Sucla2</td>\n",
       "      <td>Sucla2</td>\n",
       "      <td>20916</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K116</td>\n",
       "      <td>7.652495</td>\n",
       "      <td>1.271848</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>920</th>\n",
       "      <td>Q61425</td>\n",
       "      <td>Hydroxyacyl-coenzyme A dehydrogenase, mitochon...</td>\n",
       "      <td>Hadh</td>\n",
       "      <td>Hadh</td>\n",
       "      <td>15107</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K81</td>\n",
       "      <td>7.647936</td>\n",
       "      <td>1.228159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K414; K415</td>\n",
       "      <td>7.578536</td>\n",
       "      <td>1.372514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>803</th>\n",
       "      <td>Q80XN0</td>\n",
       "      <td>D-beta-hydroxybutyrate dehydrogenase, mitochon...</td>\n",
       "      <td>Bdh1</td>\n",
       "      <td>Bdh1</td>\n",
       "      <td>71911</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K91</td>\n",
       "      <td>7.135625</td>\n",
       "      <td>1.466350</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Accession                                               Name  Symbol  \\\n",
       "919     Q61425  Hydroxyacyl-coenzyme A dehydrogenase, mitochon...    Hadh   \n",
       "1143    Q9CZU6                    Citrate synthase, mitochondrial      Cs   \n",
       "764     Q8K2B3  Succinate dehydrogenase [ubiquinone] flavoprot...    Sdha   \n",
       "550     Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "1145    Q9CZU6                    Citrate synthase, mitochondrial      Cs   \n",
       "1142    Q9CZU6                    Citrate synthase, mitochondrial      Cs   \n",
       "616     Q03265          ATP synthase subunit alpha, mitochondrial  Atp5a1   \n",
       "836     Q8QZT1        Acetyl-CoA acetyltransferase, mitochondrial   Acat1   \n",
       "946     Q9WUR2         Enoyl-CoA delta isomerase 2, mitochondrial    Eci2   \n",
       "1093    Q99LC5  Electron transfer flavoprotein subunit alpha, ...    Etfa   \n",
       "1027    Q8BWT1             3-ketoacyl-CoA thiolase, mitochondrial   Acaa2   \n",
       "435     Q8BFR5                Elongation factor Tu, mitochondrial    Tufm   \n",
       "1136    Q9DCM2                  Glutathione S-transferase kappa 1   Gstk1   \n",
       "555     Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "416     P42125         Enoyl-CoA delta isomerase 1, mitochondrial    Eci1   \n",
       "590     P08249                Malate dehydrogenase, mitochondrial    Mdh2   \n",
       "1088    Q8K4F5                                     Protein ABHD11  Abhd11   \n",
       "626     Q03265          ATP synthase subunit alpha, mitochondrial  Atp5a1   \n",
       "1144    Q9CZU6                    Citrate synthase, mitochondrial      Cs   \n",
       "1222    P03930                             ATP synthase protein 8  Mtatp8   \n",
       "415     P42125         Enoyl-CoA delta isomerase 1, mitochondrial    Eci1   \n",
       "969     Q9Z2I9  Succinate--CoA ligase [ADP-forming] subunit be...  Sucla2   \n",
       "920     Q61425  Hydroxyacyl-coenzyme A dehydrogenase, mitochon...    Hadh   \n",
       "548     Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "803     Q80XN0  D-beta-hydroxybutyrate dehydrogenase, mitochon...    Bdh1   \n",
       "\n",
       "     Gene Symbol  Entrez  Matrix    IMS Acetylated Residues  DKOvsDFC_FC  \\\n",
       "919         Hadh   15107    True  False                K312    16.546091   \n",
       "1143          Cs   12974    True  False                K393    16.359824   \n",
       "764         Sdha   66945    True  False                K179    16.164221   \n",
       "550        Hadha   97212    True  False                K334    14.830350   \n",
       "1145          Cs   12974    True  False                 K52    14.824442   \n",
       "1142          Cs   12974    True  False            K49; K52    14.108287   \n",
       "616       Atp5a1   11946    True  False          K531; K539    13.955865   \n",
       "836        Acat1  110446    True  False                K260    13.620694   \n",
       "946         Eci2   23986    True  False                 K60    11.757587   \n",
       "1093        Etfa  110842    True  False                K226    11.376252   \n",
       "1027       Acaa2   52538    True  False                K209    11.061172   \n",
       "435         Tufm  233870    True  False                 K88    10.965843   \n",
       "1136       Gstk1   76263   False  False                 K93    10.913403   \n",
       "555        Hadha   97212    True  False                K406    10.145445   \n",
       "416         Eci1   13177    True  False                K255     9.797140   \n",
       "590         Mdh2   17448    True  False                K239     9.716530   \n",
       "1088      Abhd11   68758    True  False                K196     9.649334   \n",
       "626       Atp5a1   11946    True  False                K498     8.830434   \n",
       "1144          Cs   12974    True  False                K459     8.363489   \n",
       "1222        ATP8   17706   False  False                 K46     8.320988   \n",
       "415         Eci1   13177    True  False                K283     7.993041   \n",
       "969       Sucla2   20916    True  False                K116     7.652495   \n",
       "920         Hadh   15107    True  False                 K81     7.647936   \n",
       "548        Hadha   97212    True  False          K414; K415     7.578536   \n",
       "803         Bdh1   71911    True  False                 K91     7.135625   \n",
       "\n",
       "      CrATKOvsCrATFL_FC  \n",
       "919            1.302894  \n",
       "1143           1.418364  \n",
       "764            1.199551  \n",
       "550            1.352968  \n",
       "1145           1.795159  \n",
       "1142           1.378787  \n",
       "616            2.165616  \n",
       "836            1.284992  \n",
       "946            1.287511  \n",
       "1093           1.483240  \n",
       "1027           1.586879  \n",
       "435            1.561059  \n",
       "1136           1.171825  \n",
       "555            1.121315  \n",
       "416            2.060486  \n",
       "590            1.409407  \n",
       "1088           1.307799  \n",
       "626            1.179997  \n",
       "1144           1.307401  \n",
       "1222           1.230055  \n",
       "415            1.210816  \n",
       "969            1.271848  \n",
       "920            1.228159  \n",
       "548            1.372514  \n",
       "803            1.466350  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p1_export_cols = ['Accession', 'Name', 'Symbol', 'Gene Symbol', 'Entrez', \n",
    "                  'Matrix', 'IMS', 'Acetylated Residues', 'DKOvsDFC_FC', 'CrATKOvsCrATFL_FC']\n",
    "\n",
    "\n",
    "p1[p1_export_cols].head(25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>Name</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Entrez</th>\n",
       "      <th>Matrix</th>\n",
       "      <th>IMS</th>\n",
       "      <th>Acetylated Residues</th>\n",
       "      <th>DKOvsDFC_FC</th>\n",
       "      <th>S3KOvsS3FL_FC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>752</th>\n",
       "      <td>Q9CZU6</td>\n",
       "      <td>Citrate synthase, mitochondrial</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Cs</td>\n",
       "      <td>12974</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K52</td>\n",
       "      <td>203.430666</td>\n",
       "      <td>11.434282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>661</th>\n",
       "      <td>Q8BWT1</td>\n",
       "      <td>3-ketoacyl-CoA thiolase, mitochondrial</td>\n",
       "      <td>Acaa2</td>\n",
       "      <td>Acaa2</td>\n",
       "      <td>52538</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K209</td>\n",
       "      <td>120.487254</td>\n",
       "      <td>8.361982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>750</th>\n",
       "      <td>Q9CZU6</td>\n",
       "      <td>Citrate synthase, mitochondrial</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Cs</td>\n",
       "      <td>12974</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K393</td>\n",
       "      <td>79.247040</td>\n",
       "      <td>12.337543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>609</th>\n",
       "      <td>Q9WUR2</td>\n",
       "      <td>Enoyl-CoA delta isomerase 2, mitochondrial</td>\n",
       "      <td>Eci2</td>\n",
       "      <td>Eci2</td>\n",
       "      <td>23986</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K60</td>\n",
       "      <td>74.078830</td>\n",
       "      <td>11.796914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>P08249</td>\n",
       "      <td>Malate dehydrogenase, mitochondrial</td>\n",
       "      <td>Mdh2</td>\n",
       "      <td>Mdh2</td>\n",
       "      <td>17448</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K239</td>\n",
       "      <td>67.106543</td>\n",
       "      <td>13.180614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K259</td>\n",
       "      <td>57.811284</td>\n",
       "      <td>91.912696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>812</th>\n",
       "      <td>P03930</td>\n",
       "      <td>ATP synthase protein 8</td>\n",
       "      <td>Mtatp8</td>\n",
       "      <td>ATP8</td>\n",
       "      <td>17706</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>K46</td>\n",
       "      <td>53.756961</td>\n",
       "      <td>10.581788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K411; K413</td>\n",
       "      <td>50.224529</td>\n",
       "      <td>9.495206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K</td>\n",
       "      <td>43.299145</td>\n",
       "      <td>8.348417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>768</th>\n",
       "      <td>Q9DB20</td>\n",
       "      <td>ATP synthase subunit O, mitochondrial</td>\n",
       "      <td>Atp5o</td>\n",
       "      <td>Atp5o; LOC100047429; LOC102641678</td>\n",
       "      <td>28080</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K97</td>\n",
       "      <td>41.507593</td>\n",
       "      <td>30.187432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>P42125</td>\n",
       "      <td>Enoyl-CoA delta isomerase 1, mitochondrial</td>\n",
       "      <td>Eci1</td>\n",
       "      <td>Eci1</td>\n",
       "      <td>13177</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K283</td>\n",
       "      <td>39.648143</td>\n",
       "      <td>8.475234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>744</th>\n",
       "      <td>Q9DCT2</td>\n",
       "      <td>NADH dehydrogenase [ubiquinone] iron-sulfur pr...</td>\n",
       "      <td>Ndufs3</td>\n",
       "      <td>Ndufs3</td>\n",
       "      <td>68349</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K42</td>\n",
       "      <td>39.142628</td>\n",
       "      <td>2.874790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>708</th>\n",
       "      <td>Q8K4F5</td>\n",
       "      <td>Protein ABHD11</td>\n",
       "      <td>Abhd11</td>\n",
       "      <td>Abhd11</td>\n",
       "      <td>68758</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K</td>\n",
       "      <td>37.200521</td>\n",
       "      <td>2.740276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K406; K411</td>\n",
       "      <td>36.991018</td>\n",
       "      <td>4.769096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>Q8K2B3</td>\n",
       "      <td>Succinate dehydrogenase [ubiquinone] flavoprot...</td>\n",
       "      <td>Sdha</td>\n",
       "      <td>Sdha</td>\n",
       "      <td>66945</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K179</td>\n",
       "      <td>33.531861</td>\n",
       "      <td>28.467004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K303</td>\n",
       "      <td>32.605521</td>\n",
       "      <td>3.636055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>828</th>\n",
       "      <td>Q99KR7</td>\n",
       "      <td>Peptidyl-prolyl cis-trans isomerase F, mitocho...</td>\n",
       "      <td>Ppif</td>\n",
       "      <td>Ppif</td>\n",
       "      <td>105675</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K</td>\n",
       "      <td>31.219108</td>\n",
       "      <td>7.417711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705</th>\n",
       "      <td>P97450</td>\n",
       "      <td>ATP synthase-coupling factor 6, mitochondrial</td>\n",
       "      <td>Atp5j</td>\n",
       "      <td>Atp5j</td>\n",
       "      <td>11957</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>K94</td>\n",
       "      <td>30.950801</td>\n",
       "      <td>7.647413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K406</td>\n",
       "      <td>29.440771</td>\n",
       "      <td>11.727291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K289</td>\n",
       "      <td>27.723018</td>\n",
       "      <td>8.880307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>791</th>\n",
       "      <td>Q9DBL1</td>\n",
       "      <td>Short/branched chain specific acyl-CoA dehydro...</td>\n",
       "      <td>Acadsb</td>\n",
       "      <td>Acadsb</td>\n",
       "      <td>66885</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K284</td>\n",
       "      <td>25.533180</td>\n",
       "      <td>3.449302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>97212</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K334</td>\n",
       "      <td>24.665954</td>\n",
       "      <td>10.829101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>749</th>\n",
       "      <td>Q9CZU6</td>\n",
       "      <td>Citrate synthase, mitochondrial</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Cs</td>\n",
       "      <td>12974</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K49; K52</td>\n",
       "      <td>24.416380</td>\n",
       "      <td>2.433939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>P63038</td>\n",
       "      <td>60 kDa heat shock protein, mitochondrial</td>\n",
       "      <td>Hspd1</td>\n",
       "      <td>Hspd1</td>\n",
       "      <td>15510</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K</td>\n",
       "      <td>24.193200</td>\n",
       "      <td>5.520910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>Q03265</td>\n",
       "      <td>ATP synthase subunit alpha, mitochondrial</td>\n",
       "      <td>Atp5a1</td>\n",
       "      <td>Atp5a1</td>\n",
       "      <td>11946</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>K427</td>\n",
       "      <td>23.249167</td>\n",
       "      <td>4.780851</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Accession                                               Name  Symbol  \\\n",
       "752    Q9CZU6                    Citrate synthase, mitochondrial      Cs   \n",
       "661    Q8BWT1             3-ketoacyl-CoA thiolase, mitochondrial   Acaa2   \n",
       "750    Q9CZU6                    Citrate synthase, mitochondrial      Cs   \n",
       "609    Q9WUR2         Enoyl-CoA delta isomerase 2, mitochondrial    Eci2   \n",
       "341    P08249                Malate dehydrogenase, mitochondrial    Mdh2   \n",
       "280    Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "812    P03930                             ATP synthase protein 8  Mtatp8   \n",
       "301    Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "286    Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "768    Q9DB20              ATP synthase subunit O, mitochondrial   Atp5o   \n",
       "196    P42125         Enoyl-CoA delta isomerase 1, mitochondrial    Eci1   \n",
       "744    Q9DCT2  NADH dehydrogenase [ubiquinone] iron-sulfur pr...  Ndufs3   \n",
       "708    Q8K4F5                                     Protein ABHD11  Abhd11   \n",
       "308    Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "472    Q8K2B3  Succinate dehydrogenase [ubiquinone] flavoprot...    Sdha   \n",
       "305    Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "828    Q99KR7  Peptidyl-prolyl cis-trans isomerase F, mitocho...    Ppif   \n",
       "705    P97450      ATP synthase-coupling factor 6, mitochondrial   Atp5j   \n",
       "307    Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "282    Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "791    Q9DBL1  Short/branched chain specific acyl-CoA dehydro...  Acadsb   \n",
       "299    Q8BMS1  Trifunctional enzyme subunit alpha, mitochondrial   Hadha   \n",
       "749    Q9CZU6                    Citrate synthase, mitochondrial      Cs   \n",
       "650    P63038           60 kDa heat shock protein, mitochondrial   Hspd1   \n",
       "370    Q03265          ATP synthase subunit alpha, mitochondrial  Atp5a1   \n",
       "\n",
       "                           Gene Symbol  Entrez  Matrix    IMS  \\\n",
       "752                                 Cs   12974    True  False   \n",
       "661                              Acaa2   52538    True  False   \n",
       "750                                 Cs   12974    True  False   \n",
       "609                               Eci2   23986    True  False   \n",
       "341                               Mdh2   17448    True  False   \n",
       "280                              Hadha   97212    True  False   \n",
       "812                               ATP8   17706   False  False   \n",
       "301                              Hadha   97212    True  False   \n",
       "286                              Hadha   97212    True  False   \n",
       "768  Atp5o; LOC100047429; LOC102641678   28080    True  False   \n",
       "196                               Eci1   13177    True  False   \n",
       "744                             Ndufs3   68349    True  False   \n",
       "708                             Abhd11   68758    True  False   \n",
       "308                              Hadha   97212    True  False   \n",
       "472                               Sdha   66945    True  False   \n",
       "305                              Hadha   97212    True  False   \n",
       "828                               Ppif  105675    True  False   \n",
       "705                              Atp5j   11957   False   True   \n",
       "307                              Hadha   97212    True  False   \n",
       "282                              Hadha   97212    True  False   \n",
       "791                             Acadsb   66885    True  False   \n",
       "299                              Hadha   97212    True  False   \n",
       "749                                 Cs   12974    True  False   \n",
       "650                              Hspd1   15510    True  False   \n",
       "370                             Atp5a1   11946    True  False   \n",
       "\n",
       "    Acetylated Residues  DKOvsDFC_FC  S3KOvsS3FL_FC  \n",
       "752                 K52   203.430666      11.434282  \n",
       "661                K209   120.487254       8.361982  \n",
       "750                K393    79.247040      12.337543  \n",
       "609                 K60    74.078830      11.796914  \n",
       "341                K239    67.106543      13.180614  \n",
       "280                K259    57.811284      91.912696  \n",
       "812                 K46    53.756961      10.581788  \n",
       "301          K411; K413    50.224529       9.495206  \n",
       "286                   K    43.299145       8.348417  \n",
       "768                 K97    41.507593      30.187432  \n",
       "196                K283    39.648143       8.475234  \n",
       "744                 K42    39.142628       2.874790  \n",
       "708                   K    37.200521       2.740276  \n",
       "308          K406; K411    36.991018       4.769096  \n",
       "472                K179    33.531861      28.467004  \n",
       "305                K303    32.605521       3.636055  \n",
       "828                   K    31.219108       7.417711  \n",
       "705                 K94    30.950801       7.647413  \n",
       "307                K406    29.440771      11.727291  \n",
       "282                K289    27.723018       8.880307  \n",
       "791                K284    25.533180       3.449302  \n",
       "299                K334    24.665954      10.829101  \n",
       "749            K49; K52    24.416380       2.433939  \n",
       "650                   K    24.193200       5.520910  \n",
       "370                K427    23.249167       4.780851  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p2_export_cols = ['Accession', 'Name', 'Symbol', 'Gene Symbol', 'Entrez', \n",
    "                  'Matrix', 'IMS', 'Acetylated Residues', 'DKOvsDFC_FC', 'S3KOvsS3FL_FC']\n",
    "\n",
    "\n",
    "p2[p2_export_cols].head(25)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Export the Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write to an excel file\n",
    "writer = pd.ExcelWriter('processed_files/Williams_2019_PRX_Processed_Data.xlsx', \n",
    "                        options={'strings_to_numbers': True})\n",
    "\n",
    "p1_ro_acetyl.to_excel(writer, sheet_name='p1_DKOvsCrAT_RO_Kac', index=False)\n",
    "p2_ro_acetyl.to_excel(writer, sheet_name='p2_DKOvsSirt3_RO_Kac', index=False)\n",
    "\n",
    "p1_acetyl.to_excel(writer, sheet_name='p1_DKOvsCrAT_Kac', index=False)\n",
    "p2_acetyl.to_excel(writer, sheet_name='p2_DKOvsSirt3_Kac', index=False)\n",
    "\n",
    "p1_protein.to_excel(writer, sheet_name='p1_DKOvsCrAT_Protein', index=False)\n",
    "p2_protein.to_excel(writer, sheet_name='p2_DKOvsSirt3_Protein', index=False)\n",
    "\n",
    "writer.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write to csv files\n",
    "p1_ro_acetyl.to_csv('processed_files/p1_DKOvsCrAT_RO_Kac.csv', index=False)\n",
    "p2_ro_acetyl.to_csv('processed_files/p2_DKOvsSirt3_RO_Kac.csv', index=False)\n",
    "\n",
    "p1_acetyl.to_csv('processed_files/p1_DKOvsCrAT_Kac.csv', index=False)\n",
    "p2_acetyl.to_csv('processed_files/p2_DKOvsSirt3_Kac.csv', index=False)\n",
    "\n",
    "p1_protein.to_csv('processed_files/p1_DKOvsCrAT_Protein.csv', index=False)\n",
    "p2_protein.to_csv('processed_files/p2_DKOvsSirt3_Protein.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write to an excel file\n",
    "writer = pd.ExcelWriter('processed_files/Williams_2019_PRX_Top25.xlsx', \n",
    "                        options={'strings_to_numbers': True})\n",
    "\n",
    "(p1[p1_export_cols].sort_values('DKOvsDFC_FC', ascending=False).head(25)\n",
    " .to_excel(writer, sheet_name='p1_Top25_Kac_DKOvsDFC', index=False))\n",
    "\n",
    "(p1[p1_export_cols].sort_values('CrATKOvsCrATFL_FC', ascending=False).head(25)\n",
    " .to_excel(writer, sheet_name='p1_Top25_Kac_CrATKOvsCrATFL', index=False))\n",
    "\n",
    "(p2[p2_export_cols].sort_values('DKOvsDFC_FC', ascending=False).head(25)\n",
    " .to_excel(writer, sheet_name='p2_Top25_Kac_DKOvsDFC', index=False))\n",
    "\n",
    "(p2[p2_export_cols].sort_values('S3KOvsS3FL_FC', ascending=False).head(25)\n",
    " .to_excel(writer, sheet_name='p2_Top25_Kac_S3KOvsS3FL', index=False))\n",
    "\n",
    "writer.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write to csv files\n",
    "(p1[p1_export_cols].sort_values('DKOvsDFC_FC', ascending=False).head(25)\n",
    " .to_csv('processed_files/p1_Top25_Kac_DKOvsDFC.csv', index=False))\n",
    "\n",
    "(p1[p1_export_cols].sort_values('CrATKOvsCrATFL_FC', ascending=False).head(25)\n",
    " .to_csv('processed_files/p1_Top25_Kac_CrATKOvsCrATFL.csv', index=False))\n",
    "\n",
    "(p2[p2_export_cols].sort_values('DKOvsDFC_FC', ascending=False).head(25)\n",
    " .to_csv('processed_files/p2_Top25_Kac_DKOvsDFC.csv', index=False))\n",
    "\n",
    "(p2[p2_export_cols].sort_values('S3KOvsS3FL_FC', ascending=False)\n",
    " .head(25).to_csv('processed_files/p2_Top25_Kac_S3KOvsS3FL.csv', index=False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
