{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import the processed data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "ack1 = pd.read_csv('processed_files\\p1_DKOvsCrAT_RO_Kac.csv')\n",
    "ack2 = pd.read_csv('processed_files\\p2_DKOvsSirt3_RO_Kac.csv')\n",
    "\n",
    "pro1 = pd.read_csv('processed_files\\p1_DKOvsCrAT_Protein.csv')\n",
    "pro2 = pd.read_csv('processed_files\\p2_DKOvsSirt3_Protein.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Description</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Entrez</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Positions in Proteins</th>\n",
       "      <th>Modifications in Proteins</th>\n",
       "      <th>Modifications</th>\n",
       "      <th>...</th>\n",
       "      <th>DKOvsDFC_p_value</th>\n",
       "      <th>CrATKOvsCrATFL_p_value</th>\n",
       "      <th>DKOvsCrATKO_p_value</th>\n",
       "      <th>DFCvsCrATFL_p_value</th>\n",
       "      <th>F</th>\n",
       "      <th>F.p.value</th>\n",
       "      <th>DKOvsDFC_significant</th>\n",
       "      <th>CrATKOvsCrATFL_significant</th>\n",
       "      <th>DKOvsCrATKO_significant</th>\n",
       "      <th>DFCvsCrATFL_significant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Hydroxyacyl-coenzyme A dehydrogenase, mitochon...</td>\n",
       "      <td>Hadh</td>\n",
       "      <td>Hadh</td>\n",
       "      <td>Hydroxyacyl-coenzyme A dehydrogenase, mitochon...</td>\n",
       "      <td>Q61425</td>\n",
       "      <td>15107.0</td>\n",
       "      <td>TGEGFYKYK</td>\n",
       "      <td>Q61425 [306-314]</td>\n",
       "      <td>Q61425 1xAcetyl [K312]</td>\n",
       "      <td>1xAcetyl [K7]; 1xTMT6plex [K9]; 1xTMT6plex [N-...</td>\n",
       "      <td>...</td>\n",
       "      <td>1.267950e-09</td>\n",
       "      <td>0.034555</td>\n",
       "      <td>1.355804e-10</td>\n",
       "      <td>0.000235</td>\n",
       "      <td>324.124630</td>\n",
       "      <td>3.071493e-10</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Citrate synthase, mitochondrial</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Citrate synthase, mitochondrial OS=Mus musculu...</td>\n",
       "      <td>Q9CZU6</td>\n",
       "      <td>12974.0</td>\n",
       "      <td>IVPNILLEQGKAK</td>\n",
       "      <td>Q9CZU6 [383-395]</td>\n",
       "      <td>Q9CZU6 1xAcetyl [K393]</td>\n",
       "      <td>1xAcetyl [K11]; 1xTMT6plex [K13]; 1xTMT6plex [...</td>\n",
       "      <td>...</td>\n",
       "      <td>3.025312e-09</td>\n",
       "      <td>0.014144</td>\n",
       "      <td>2.612986e-09</td>\n",
       "      <td>0.297806</td>\n",
       "      <td>199.929464</td>\n",
       "      <td>3.305818e-09</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Succinate dehydrogenase [ubiquinone] flavoprot...</td>\n",
       "      <td>Sdha</td>\n",
       "      <td>Sdha</td>\n",
       "      <td>Succinate dehydrogenase [ubiquinone] flavoprot...</td>\n",
       "      <td>Q8K2B3</td>\n",
       "      <td>66945.0</td>\n",
       "      <td>AFGGQSLKFGK</td>\n",
       "      <td>Q8K2B3 [172-182]; A0A1Y7VJ55 [170-180]</td>\n",
       "      <td>Q8K2B3 1xAcetyl [K179]; A0A1Y7VJ55 1xAcetyl [K...</td>\n",
       "      <td>1xAcetyl [K8]; 1xTMT6plex [K11]; 1xTMT6plex [N...</td>\n",
       "      <td>...</td>\n",
       "      <td>5.864913e-10</td>\n",
       "      <td>0.096459</td>\n",
       "      <td>4.282730e-10</td>\n",
       "      <td>0.845245</td>\n",
       "      <td>271.052008</td>\n",
       "      <td>7.414188e-10</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondr...</td>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>97212.0</td>\n",
       "      <td>FGELALTKESK</td>\n",
       "      <td>Q8BMS1 [327-337]</td>\n",
       "      <td>Q8BMS1 1xAcetyl [K334]</td>\n",
       "      <td>1xAcetyl [K8]; 1xTMT6plex [K11]; 1xTMT6plex [N...</td>\n",
       "      <td>...</td>\n",
       "      <td>5.235680e-11</td>\n",
       "      <td>0.002444</td>\n",
       "      <td>1.476519e-11</td>\n",
       "      <td>0.000725</td>\n",
       "      <td>533.297696</td>\n",
       "      <td>2.618139e-11</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Citrate synthase, mitochondrial</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Cs</td>\n",
       "      <td>Citrate synthase, mitochondrial OS=Mus musculu...</td>\n",
       "      <td>Q9CZU6</td>\n",
       "      <td>12974.0</td>\n",
       "      <td>TFKQQHGK</td>\n",
       "      <td>Q9CZU6 [50-57]</td>\n",
       "      <td>Q9CZU6 1xAcetyl [K52]</td>\n",
       "      <td>1xAcetyl [K3]; 1xTMT6plex [K8]; 1xTMT6plex [N-...</td>\n",
       "      <td>...</td>\n",
       "      <td>9.560681e-10</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>5.296337e-10</td>\n",
       "      <td>0.001231</td>\n",
       "      <td>296.660866</td>\n",
       "      <td>4.752772e-10</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 72 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Name Symbol Gene Symbol  \\\n",
       "0  Hydroxyacyl-coenzyme A dehydrogenase, mitochon...   Hadh        Hadh   \n",
       "1                    Citrate synthase, mitochondrial     Cs          Cs   \n",
       "2  Succinate dehydrogenase [ubiquinone] flavoprot...   Sdha        Sdha   \n",
       "3  Trifunctional enzyme subunit alpha, mitochondrial  Hadha       Hadha   \n",
       "4                    Citrate synthase, mitochondrial     Cs          Cs   \n",
       "\n",
       "                                         Description Accession   Entrez  \\\n",
       "0  Hydroxyacyl-coenzyme A dehydrogenase, mitochon...    Q61425  15107.0   \n",
       "1  Citrate synthase, mitochondrial OS=Mus musculu...    Q9CZU6  12974.0   \n",
       "2  Succinate dehydrogenase [ubiquinone] flavoprot...    Q8K2B3  66945.0   \n",
       "3  Trifunctional enzyme subunit alpha, mitochondr...    Q8BMS1  97212.0   \n",
       "4  Citrate synthase, mitochondrial OS=Mus musculu...    Q9CZU6  12974.0   \n",
       "\n",
       "        Sequence                   Positions in Proteins  \\\n",
       "0      TGEGFYKYK                        Q61425 [306-314]   \n",
       "1  IVPNILLEQGKAK                        Q9CZU6 [383-395]   \n",
       "2    AFGGQSLKFGK  Q8K2B3 [172-182]; A0A1Y7VJ55 [170-180]   \n",
       "3    FGELALTKESK                        Q8BMS1 [327-337]   \n",
       "4       TFKQQHGK                          Q9CZU6 [50-57]   \n",
       "\n",
       "                           Modifications in Proteins  \\\n",
       "0                             Q61425 1xAcetyl [K312]   \n",
       "1                             Q9CZU6 1xAcetyl [K393]   \n",
       "2  Q8K2B3 1xAcetyl [K179]; A0A1Y7VJ55 1xAcetyl [K...   \n",
       "3                             Q8BMS1 1xAcetyl [K334]   \n",
       "4                              Q9CZU6 1xAcetyl [K52]   \n",
       "\n",
       "                                       Modifications  ...  DKOvsDFC_p_value  \\\n",
       "0  1xAcetyl [K7]; 1xTMT6plex [K9]; 1xTMT6plex [N-...  ...      1.267950e-09   \n",
       "1  1xAcetyl [K11]; 1xTMT6plex [K13]; 1xTMT6plex [...  ...      3.025312e-09   \n",
       "2  1xAcetyl [K8]; 1xTMT6plex [K11]; 1xTMT6plex [N...  ...      5.864913e-10   \n",
       "3  1xAcetyl [K8]; 1xTMT6plex [K11]; 1xTMT6plex [N...  ...      5.235680e-11   \n",
       "4  1xAcetyl [K3]; 1xTMT6plex [K8]; 1xTMT6plex [N-...  ...      9.560681e-10   \n",
       "\n",
       "   CrATKOvsCrATFL_p_value  DKOvsCrATKO_p_value  DFCvsCrATFL_p_value  \\\n",
       "0                0.034555         1.355804e-10             0.000235   \n",
       "1                0.014144         2.612986e-09             0.297806   \n",
       "2                0.096459         4.282730e-10             0.845245   \n",
       "3                0.002444         1.476519e-11             0.000725   \n",
       "4                0.000176         5.296337e-10             0.001231   \n",
       "\n",
       "            F     F.p.value  DKOvsDFC_significant  CrATKOvsCrATFL_significant  \\\n",
       "0  324.124630  3.071493e-10                   1.0                         0.0   \n",
       "1  199.929464  3.305818e-09                   1.0                         1.0   \n",
       "2  271.052008  7.414188e-10                   1.0                         0.0   \n",
       "3  533.297696  2.618139e-11                   1.0                         1.0   \n",
       "4  296.660866  4.752772e-10                   1.0                         1.0   \n",
       "\n",
       "   DKOvsCrATKO_significant  DFCvsCrATFL_significant  \n",
       "0                      1.0                      0.0  \n",
       "1                      1.0                      0.0  \n",
       "2                      1.0                      0.0  \n",
       "3                      1.0                      0.0  \n",
       "4                      1.0                      0.0  \n",
       "\n",
       "[5 rows x 72 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ack1.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Making the sausage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is a hacky bit of code to create a multi-level column header for excel output; it is not very efficient or pretty."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Acetylpeptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make lists of the columns we will want to export in the AcK peptides\n",
    "ack_protein_info_list = ['Gene Symbol',\n",
    "                         'Description',\n",
    "                         'Accession', \n",
    "                         'Entrez',]\n",
    "\n",
    "ack_peptide_info_list = ['Sequence',\n",
    "                         'Positions in Proteins',\n",
    "                         'Modifications in Proteins',\n",
    "                         'Modifications', ]\n",
    "\n",
    "mitocarta_list = ['MitoCarta2_List',\n",
    "                  'Matrix',\n",
    "                  'IMS',]\n",
    "\n",
    "ack_col_list1 = [ack_protein_info_list, \n",
    "                 ack_peptide_info_list, \n",
    "                 mitocarta_list]\n",
    "\n",
    "ack_col_list2 = ack_col_list1.copy()\n",
    "\n",
    "\n",
    "# go through the RO dataframe and extract the important quantification and limma analysis columns\n",
    "for i, df in enumerate([ack1, ack2]):\n",
    "    # the column endings we care about\n",
    "    for match in ['_log2$', '_ro$', '_Log2FC$', 'p_value$', '_significant$']:\n",
    "        temp = df.filter(regex=match).columns.tolist()\n",
    "        if i==0:\n",
    "            ack_col_list1.append(temp)\n",
    "        else:\n",
    "            ack_col_list2.append(temp)\n",
    "            \n",
    "                \n",
    "# generate a group label for the various columns\n",
    "ack_col_keys = ['Protein', 'Peptide', 'MitoCarta 2.0 Status', \n",
    "                'Skeletal Muscle Acetylpeptide Relative Abundance (log2)', \n",
    "                'Skeletal Muscle Acetylpeptide Relative Occupancy (log2)']\n",
    "\n",
    "# account for the 3 components to grab from the limma analysis\n",
    "ack_col_keys = ack_col_keys + ['limma Analysis on Relative Occupancy']*3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove un-wanted columns\n",
    "ack1 = ack1[np.hstack(ack_col_list1).tolist()]\n",
    "ack2 = ack2[np.hstack(ack_col_list2).tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Miniconda3\\lib\\site-packages\\ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "C:\\ProgramData\\Miniconda3\\lib\\site-packages\\ipykernel_launcher.py:27: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    }
   ],
   "source": [
    "# hack to apply the group label to the dataframe columns for peptides\n",
    "\n",
    "#### do plex1 \n",
    "col_stack = pd.DataFrame(data=ack1.columns, columns=['headers'])\n",
    "col_stack['group'] = np.nan\n",
    "\n",
    "for cols, key in zip(ack_col_list1, ack_col_keys):\n",
    "    col_stack.group[col_stack.headers.isin(cols)] = key\n",
    "    \n",
    "# prep for and then concat.\n",
    "col_stack = col_stack.T\n",
    "col_stack.columns = col_stack.loc['headers']\n",
    "\n",
    "ack1 = pd.concat([ack1, col_stack])\n",
    "\n",
    "# generate new columns, aligned to the df, from the dataframe\n",
    "ack1.columns = pd.MultiIndex.from_arrays(ack1.loc[['group', 'headers']].values)\n",
    "\n",
    "# get rid of the redundant rows \n",
    "ack1.drop(labels=['group', 'headers'], inplace=True)\n",
    "\n",
    "#### do plex2 \n",
    "col_stack = pd.DataFrame(data=ack2.columns, columns=['headers'])\n",
    "col_stack['group'] = np.nan\n",
    "\n",
    "for cols, key in zip(ack_col_list2, ack_col_keys):\n",
    "    col_stack.group[col_stack.headers.isin(cols)] = key\n",
    "    \n",
    "# prep for and then concat.\n",
    "col_stack = col_stack.T\n",
    "col_stack.columns = col_stack.loc['headers']\n",
    "\n",
    "ack2 = pd.concat([ack2, col_stack])\n",
    "\n",
    "# generate new columns, aligned to the df, from the dataframe\n",
    "ack2.columns = pd.MultiIndex.from_arrays(ack2.loc[['group', 'headers']].values)\n",
    "\n",
    "# get rid of the redundant rows \n",
    "ack2.drop(labels=['group', 'headers'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make the names less cluttered\n",
    "\n",
    "ra_match = '(?P<geno>.+)_(?P<tmt>.+)_\\d_Acetyl_norm_log2$'\n",
    "ro_match = '(?P<geno>.+)_(?P<tmt>.+)_\\d_Acetyl_norm_log2_ro$'\n",
    "ack_replace = '\\g<geno>_\\g<tmt>'\n",
    "\n",
    "for match in [ra_match, ro_match]:\n",
    "    ack1.rename(columns=lambda x: re.sub(match, ack_replace, x),\n",
    "                level=1, \n",
    "                inplace=True)\n",
    "\n",
    "    ack2.rename(columns=lambda x: re.sub(match, ack_replace, x),\n",
    "                level=1, \n",
    "                inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make lists of the columns we will want to export in the Proteins\n",
    "pro_protein_info_list = ['Gene Symbol',\n",
    "                         'Description',\n",
    "                         'Accession',\n",
    "                         'Entrez',\n",
    "                         'Ensembl',\n",
    "                         'Chromosome']\n",
    "\n",
    "pro_id_stats_info_list = ['Coverage [%]',\n",
    "                          '# Peptides',\n",
    "                          '# PSMs',\n",
    "                          '# Protein Unique Peptides',\n",
    "                          '# Unique Peptides', \n",
    "                          'Acetyl (K) Count', \n",
    "                          'Acetyl (K) Positions']\n",
    "\n",
    "pro_col_list1 = [pro_protein_info_list, \n",
    "                 mitocarta_list, \n",
    "                 pro_id_stats_info_list]\n",
    "\n",
    "pro_col_list2 = pro_col_list1.copy()\n",
    "\n",
    "\n",
    "# go through the Protein Dataframe and extract the relevant quantification information\n",
    "for i, df in enumerate([pro1, pro2]):\n",
    "    for match in ['_log2$', '_Log2FC$', 'p_value$', '_significant$']:\n",
    "        temp = df.filter(regex=match).columns.tolist()\n",
    "    \n",
    "        if i==0:\n",
    "            pro_col_list1.append(temp)\n",
    "        else:\n",
    "            pro_col_list2.append(temp)\n",
    "\n",
    "# generate a group label for the various columns\n",
    "pro_col_keys = ['Protein', 'MitoCarta 2.0 Status', \n",
    "                'Identification Statistics',\n",
    "                'Skeletal Muscle Protein Relative Abundance (log2)']\n",
    "\n",
    "# account for the 3 components to grab from the limma analysis\n",
    "pro_col_keys = pro_col_keys + ['limma Analysis on Protein Relative Abundance']*3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "pro1 = pro1[np.hstack(pro_col_list1).tolist()].copy()\n",
    "pro2 = pro2[np.hstack(pro_col_list2).tolist()].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Miniconda3\\lib\\site-packages\\ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "C:\\ProgramData\\Miniconda3\\lib\\site-packages\\ipykernel_launcher.py:27: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    }
   ],
   "source": [
    "# hack to apply the group label to the dataframe columns for peptides\n",
    "\n",
    "#### do plex1 \n",
    "col_stack = pd.DataFrame(data=pro1.columns, columns=['headers'])\n",
    "col_stack['group'] = np.nan\n",
    "\n",
    "for cols, key in zip(pro_col_list1, pro_col_keys):\n",
    "    col_stack.group[col_stack.headers.isin(cols)] = key\n",
    "    \n",
    "# prep for and then concat.\n",
    "col_stack = col_stack.T\n",
    "col_stack.columns = col_stack.loc['headers']\n",
    "\n",
    "pro1 = pd.concat([pro1, col_stack])\n",
    "\n",
    "# generate new columns, aligned to the df, from the dataframe\n",
    "pro1.columns = pd.MultiIndex.from_arrays(pro1.loc[['group', 'headers']].values)\n",
    "\n",
    "# get rid of the redundant rows \n",
    "pro1.drop(labels=['group', 'headers'], inplace=True)\n",
    "\n",
    "#### do plex2\n",
    "col_stack = pd.DataFrame(data=pro2.columns, columns=['headers'])\n",
    "col_stack['group'] = np.nan\n",
    "\n",
    "for cols, key in zip(pro_col_list2, pro_col_keys):\n",
    "    col_stack.group[col_stack.headers.isin(cols)] = key\n",
    "    \n",
    "# prep for and then concat.\n",
    "col_stack = col_stack.T\n",
    "col_stack.columns = col_stack.loc['headers']\n",
    "\n",
    "pro2 = pd.concat([pro2, col_stack])\n",
    "\n",
    "# generate new columns, aligned to the df, from the dataframe\n",
    "pro2.columns = pd.MultiIndex.from_arrays(pro2.loc[['group', 'headers']].values)\n",
    "\n",
    "# get rid of the redundant rows \n",
    "pro2.drop(labels=['group', 'headers'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make the names less cluttered\n",
    "\n",
    "ra_match = '(?P<geno>.+)_(?P<tmt>.+)_\\d_Proteins_norm_log2$'\n",
    "pro_replace = '\\g<geno>_\\g<tmt>'\n",
    "\n",
    "pro1.rename(columns=lambda x: re.sub(ra_match, pro_replace, x),\n",
    "            level=1, \n",
    "            inplace=True)\n",
    "\n",
    "pro2.rename(columns=lambda x: re.sub(ra_match, pro_replace, x),\n",
    "            level=1, \n",
    "            inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# export to excel \n",
    "writer = pd.ExcelWriter('processed_files/SKM DKO 2x10plex_RO_limma.xlsx', \n",
    "                        options={'strings_to_numbers': True}, \n",
    "                        engine='xlsxwriter')\n",
    "\n",
    "ack1.to_excel(writer, sheet_name='Plex1 Acetylpeptide SUMMARY')\n",
    "ack2.to_excel(writer, sheet_name='Plex2 Acetylpeptide SUMMARY')\n",
    "\n",
    "pro1.to_excel(writer, sheet_name='Plex1 Protein SUMMARY')\n",
    "pro2.to_excel(writer, sheet_name='Plex2 Protein SUMMARY')\n",
    "\n",
    "# to include the proteome discoverer 2.2 exported data,\n",
    "# uncomment the following code:\n",
    "\n",
    "# peptides_file = 'pd22_exports/pd22_export_Williams_2019_PeptideGroups.txt'\n",
    "# proteins_file = 'pd22_exports/pd22_export_Williams_2019_Proteins.txt'\n",
    "# pep_pd22 = pd.read_csv(peptides_file, sep='\\t', low_memory=False)\n",
    "# pro_pd22 = pd.read_csv(proteins_file, sep='\\t', low_memory=False)\n",
    "# pep_pd22.to_excel(writer, sheet_name='Peptide Groups RAW', index=False)\n",
    "# pro_pd22.to_excel(writer, sheet_name='Proteins RAW', index=False)\n",
    "\n",
    "writer.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
